Upstream-Status: Backport

Signed-off-by: Maksim Sisov <msisov@igalia.com>
---
From 486645f736c8bd24b477049006b6af71a5df28cf Mon Sep 17 00:00:00 2001
From: Michael Spang <spang@chromium.org>
Date: Thu, 21 Mar 2019 21:10:25 +0000
Subject: [PATCH 13/27] ozone: Include <memory> and <utility> where necessary

The code review server is now complaining about this, so try to fix it
throughout //ui/ozone.

Change-Id: I5fff8906fc2d035287ffcf5e9092ff6f25778107
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/1526783
Commit-Queue: Michael Spang <spang@chromium.org>
Reviewed-by: Sergey Ulanov <sergeyu@chromium.org>
Reviewed-by: Maksim Sisov <msisov@igalia.com>
Cr-Commit-Position: refs/heads/master@{#643115}
---
 ui/ozone/common/linux/gbm_device.h                    |  5 +++--
 ui/ozone/common/linux/gbm_wrapper.cc                  | 11 +++++++----
 ui/ozone/common/stub_client_native_pixmap_factory.cc  |  2 ++
 ui/ozone/common/stub_overlay_manager.cc               |  3 +++
 ui/ozone/common/stub_overlay_manager.h                |  2 ++
 ui/ozone/demo/demo_window.cc                          |  2 ++
 ui/ozone/demo/demo_window.h                           |  2 ++
 ui/ozone/demo/gl_renderer.cc                          |  3 +++
 ui/ozone/demo/gl_renderer.h                           |  2 ++
 ui/ozone/demo/ozone_demo.cc                           |  1 +
 ui/ozone/demo/renderer_factory.h                      |  2 ++
 ui/ozone/demo/simple_renderer_factory.cc              |  1 +
 ui/ozone/demo/simple_renderer_factory.h               |  2 ++
 ui/ozone/demo/skia/skia_demo.cc                       |  1 +
 ui/ozone/demo/skia/skia_gl_renderer.cc                |  3 +++
 ui/ozone/demo/skia/skia_gl_renderer.h                 |  2 ++
 ui/ozone/demo/skia/skia_renderer_factory.cc           |  1 +
 ui/ozone/demo/skia/skia_renderer_factory.h            |  2 ++
 ui/ozone/demo/skia/skia_surfaceless_gl_renderer.cc    |  2 ++
 ui/ozone/demo/software_renderer.cc                    |  3 +++
 ui/ozone/demo/surfaceless_gl_renderer.cc              |  2 ++
 ui/ozone/demo/vulkan_overlay_renderer.cc              |  2 ++
 ui/ozone/demo/vulkan_overlay_renderer.h               |  1 +
 ui/ozone/demo/vulkan_renderer.cc                      |  2 ++
 ui/ozone/demo/vulkan_renderer.h                       |  1 +
 ui/ozone/demo/window_manager.cc                       |  3 +++
 ui/ozone/demo/window_manager.h                        |  1 +
 ui/ozone/gl/gl_image_ozone_native_pixmap_unittest.cc  |  1 -
 .../cast/client_native_pixmap_factory_cast.cc         |  2 ++
 ui/ozone/platform/cast/gl_ozone_egl_cast.cc           |  2 +-
 ui/ozone/platform/cast/gl_ozone_egl_cast.h            |  1 -
 ui/ozone/platform/cast/gl_surface_cast.cc             |  1 +
 ui/ozone/platform/cast/gl_surface_cast.h              |  1 +
 ui/ozone/platform/cast/overlay_manager_cast.cc        |  2 ++
 ui/ozone/platform/cast/ozone_platform_cast.cc         |  1 +
 ui/ozone/platform/cast/surface_factory_cast.cc        |  1 +
 ui/ozone/platform/drm/common/display_types.h          |  4 +++-
 ui/ozone/platform/drm/common/drm_overlay_manager.cc   |  1 +
 ui/ozone/platform/drm/common/drm_util.cc              |  1 +
 ui/ozone/platform/drm/common/drm_util.h               |  1 +
 ui/ozone/platform/drm/gpu/crtc_controller.cc          |  2 ++
 ui/ozone/platform/drm/gpu/drm_device.cc               |  1 +
 ui/ozone/platform/drm/gpu/drm_device.h                |  2 +-
 ui/ozone/platform/drm/gpu/drm_device_manager.cc       |  1 +
 ui/ozone/platform/drm/gpu/drm_display.cc              |  1 +
 ui/ozone/platform/drm/gpu/drm_display.h               |  2 +-
 ui/ozone/platform/drm/gpu/drm_framebuffer.cc          |  2 ++
 ui/ozone/platform/drm/gpu/drm_gpu_display_manager.cc  |  2 ++
 ui/ozone/platform/drm/gpu/drm_gpu_display_manager.h   |  1 -
 ui/ozone/platform/drm/gpu/drm_overlay_plane.cc        |  2 ++
 ui/ozone/platform/drm/gpu/drm_overlay_plane.h         |  1 +
 ui/ozone/platform/drm/gpu/drm_overlay_validator.cc    |  2 ++
 .../drm/gpu/drm_overlay_validator_unittest.cc         |  1 -
 ui/ozone/platform/drm/gpu/drm_thread.cc               |  1 +
 ui/ozone/platform/drm/gpu/drm_thread.h                |  1 -
 ui/ozone/platform/drm/gpu/drm_thread_message_proxy.cc |  2 ++
 ui/ozone/platform/drm/gpu/drm_thread_proxy.cc         |  3 +++
 ui/ozone/platform/drm/gpu/drm_window.cc               |  1 +
 ui/ozone/platform/drm/gpu/drm_window.h                |  1 +
 ui/ozone/platform/drm/gpu/drm_window_proxy.cc         |  2 ++
 ui/ozone/platform/drm/gpu/drm_window_unittest.cc      |  1 -
 ui/ozone/platform/drm/gpu/gbm_overlay_surface.cc      |  2 ++
 ui/ozone/platform/drm/gpu/gbm_pixmap.cc               |  2 ++
 ui/ozone/platform/drm/gpu/gbm_pixmap.h                |  1 +
 ui/ozone/platform/drm/gpu/gbm_surface_factory.cc      |  2 +-
 ui/ozone/platform/drm/gpu/gbm_surface_factory.h       |  1 -
 ui/ozone/platform/drm/gpu/gbm_surfaceless.cc          |  1 +
 .../platform/drm/gpu/hardware_display_controller.cc   |  1 +
 .../platform/drm/gpu/hardware_display_controller.h    |  1 -
 .../drm/gpu/hardware_display_controller_unittest.cc   |  2 ++
 .../drm/gpu/hardware_display_plane_manager.cc         |  2 +-
 .../platform/drm/gpu/hardware_display_plane_manager.h |  1 +
 .../drm/gpu/hardware_display_plane_manager_atomic.cc  |  2 ++
 .../drm/gpu/hardware_display_plane_manager_atomic.h   |  1 +
 .../drm/gpu/hardware_display_plane_manager_legacy.cc  |  2 ++
 .../drm/gpu/hardware_display_plane_manager_legacy.h   |  1 +
 .../gpu/hardware_display_plane_manager_unittest.cc    |  2 +-
 ui/ozone/platform/drm/gpu/mock_drm_device.cc          |  2 ++
 ui/ozone/platform/drm/gpu/mock_drm_device.h           |  2 +-
 ui/ozone/platform/drm/gpu/mock_gbm_device.cc          |  2 ++
 ui/ozone/platform/drm/gpu/page_flip_request.cc        |  2 ++
 ui/ozone/platform/drm/gpu/proxy_helpers.cc            |  2 ++
 ui/ozone/platform/drm/gpu/proxy_helpers.h             |  2 ++
 ui/ozone/platform/drm/gpu/proxy_helpers_unittest.cc   |  3 +++
 ui/ozone/platform/drm/gpu/screen_manager.cc           |  2 +-
 ui/ozone/platform/drm/gpu/screen_manager.h            |  2 +-
 ui/ozone/platform/drm/gpu/screen_manager_unittest.cc  |  2 +-
 .../platform/drm/gpu/vulkan_implementation_gbm.cc     |  2 ++
 ui/ozone/platform/drm/host/drm_cursor.cc              |  3 +++
 ui/ozone/platform/drm/host/drm_device_connector.cc    |  2 ++
 ui/ozone/platform/drm/host/drm_display_host.cc        |  3 +++
 .../platform/drm/host/drm_display_host_manager.cc     |  2 +-
 ui/ozone/platform/drm/host/drm_display_host_manager.h |  1 -
 .../drm/host/drm_gpu_platform_support_host.cc         |  1 +
 .../platform/drm/host/drm_native_display_delegate.cc  |  2 ++
 ui/ozone/platform/drm/host/host_cursor_proxy.cc       |  2 ++
 ui/ozone/platform/drm/host/host_drm_device.cc         |  3 +++
 ui/ozone/platform/drm/host/host_drm_device.h          |  2 ++
 ui/ozone/platform/drm/ozone_platform_gbm.cc           |  1 -
 .../headless/headless_native_display_delegate.cc      |  3 +++
 .../headless/headless_native_display_delegate.h       |  2 ++
 .../platform/headless/headless_surface_factory.cc     |  2 ++
 ui/ozone/platform/headless/ozone_platform_headless.cc |  2 ++
 ui/ozone/platform/magma/ozone_platform_magma.cc       |  2 ++
 ui/ozone/platform/scenic/scenic_gpu_host.cc           |  1 +
 ui/ozone/platform/scenic/scenic_surface_factory.cc    |  1 +
 ui/ozone/platform/scenic/scenic_window_canvas.cc      |  2 ++
 ui/ozone/platform/scenic/scenic_window_canvas.h       |  1 +
 ui/ozone/platform/scenic/scenic_window_manager.cc     |  2 ++
 .../platform/scenic/vulkan_implementation_scenic.cc   |  1 +
 ui/ozone/platform/wayland/gl_surface_wayland.cc       |  2 +-
 ui/ozone/platform/wayland/gpu/gbm_pixmap_wayland.cc   |  1 +
 ui/ozone/platform/wayland/gpu/gbm_pixmap_wayland.h    |  1 +
 .../platform/wayland/gpu/gbm_surfaceless_wayland.cc   |  2 ++
 .../platform/wayland/gpu/gbm_surfaceless_wayland.h    |  2 ++
 .../platform/wayland/gpu/wayland_canvas_surface.cc    |  1 +
 .../platform/wayland/gpu/wayland_connection_proxy.h   |  2 ++
 .../wayland/test/test_wayland_server_thread.cc        |  1 +
 .../wayland/test/test_wayland_server_thread.h         |  1 -
 ui/ozone/platform/wayland/wayland_buffer_fuzzer.cc    |  1 -
 ui/ozone/platform/wayland/wayland_buffer_manager.cc   |  1 +
 ui/ozone/platform/wayland/wayland_buffer_manager.h    |  1 +
 ui/ozone/platform/wayland/wayland_connection.cc       |  1 +
 ui/ozone/platform/wayland/wayland_cursor.cc           |  3 +++
 ui/ozone/platform/wayland/wayland_data_device.cc      |  1 +
 .../platform/wayland/wayland_data_device_unittest.cc  |  2 ++
 .../wayland/wayland_input_method_context_factory.cc   |  2 ++
 .../wayland/wayland_input_method_context_factory.h    |  2 ++
 .../wayland/wayland_input_method_context_unittest.cc  |  1 +
 .../platform/wayland/wayland_keyboard_unittest.cc     |  1 +
 ui/ozone/platform/wayland/wayland_object.h            |  1 -
 ui/ozone/platform/wayland/wayland_output_manager.cc   |  2 ++
 ui/ozone/platform/wayland/wayland_pointer.h           |  2 ++
 ui/ozone/platform/wayland/wayland_pointer_unittest.cc |  1 +
 ui/ozone/platform/wayland/wayland_screen_unittest.cc  |  1 -
 ui/ozone/platform/wayland/wayland_surface_factory.h   |  2 ++
 .../wayland/wayland_surface_factory_unittest.cc       |  1 +
 ui/ozone/platform/wayland/wayland_test.h              |  2 ++
 ui/ozone/platform/wayland/wayland_touch_unittest.cc   |  1 +
 ui/ozone/platform/wayland/wayland_window.cc           |  3 +--
 ui/ozone/platform/wayland/wayland_window.h            |  1 +
 ui/ozone/platform/wayland/xdg_popup_wrapper_v6.cc     |  1 +
 ui/ozone/platform/wayland/xdg_popup_wrapper_v6.h      |  2 ++
 ui/ozone/platform/windows/ozone_platform_windows.cc   |  2 ++
 ui/ozone/platform/x11/x11_screen_ozone.h              |  1 +
 ui/ozone/platform/x11/x11_screen_ozone_unittest.cc    |  2 ++
 ui/ozone/platform/x11/x11_surface_factory.cc          |  2 ++
 ui/ozone/platform/x11/x11_window_ozone_unittest.cc    |  2 ++
 ui/ozone/platform_object_internal.h                   |  2 ++
 ui/ozone/public/client_native_pixmap_factory_ozone.cc |  2 ++
 ui/ozone/public/client_native_pixmap_factory_ozone.h  |  2 ++
 ui/ozone/public/input_controller.cc                   |  2 ++
 ui/ozone/public/overlay_plane.cc                      |  2 ++
 ui/ozone/public/overlay_plane.h                       |  2 ++
 ui/ozone/public/ozone_platform.cc                     |  1 +
 ui/ozone/public/surface_factory_ozone.cc              |  1 +
 ui/ozone/public/surface_factory_ozone.h               |  1 -
 ui/ozone/public/swap_completion_callback.h            |  1 +
 158 files changed, 241 insertions(+), 36 deletions(-)

diff --git a/ui/ozone/common/linux/gbm_device.h b/ui/ozone/common/linux/gbm_device.h
index 29d9e0ab630c..26402f1505e2 100644
--- a/ui/ozone/common/linux/gbm_device.h
+++ b/ui/ozone/common/linux/gbm_device.h
@@ -5,13 +5,14 @@
 #ifndef UI_OZONE_COMMON_LINUX_GBM_DEVICE_H_
 #define UI_OZONE_COMMON_LINUX_GBM_DEVICE_H_
 
+#include <gbm.h>
+#include <memory>
+
 #include "base/files/file.h"
 #include "base/macros.h"
 #include "ui/gfx/geometry/size.h"
 #include "ui/gfx/native_pixmap_handle.h"
 
-#include <gbm.h>
-
 namespace ui {
 
 class GbmBuffer;
diff --git a/ui/ozone/common/linux/gbm_wrapper.cc b/ui/ozone/common/linux/gbm_wrapper.cc
index ce05d2930aea..a3313f14652b 100644
--- a/ui/ozone/common/linux/gbm_wrapper.cc
+++ b/ui/ozone/common/linux/gbm_wrapper.cc
@@ -5,10 +5,8 @@
 #include "ui/ozone/common/linux/gbm_wrapper.h"
 
 #include <gbm.h>
-#if !defined(MINIGBM)
-#include <fcntl.h>
-#include <xf86drm.h>
-#endif
+#include <memory>
+#include <utility>
 
 #include "base/posix/eintr_wrapper.h"
 #include "third_party/skia/include/core/SkSurface.h"
@@ -17,6 +15,11 @@
 #include "ui/ozone/common/linux/gbm_buffer.h"
 #include "ui/ozone/common/linux/gbm_device.h"
 
+#if !defined(MINIGBM)
+#include <fcntl.h>
+#include <xf86drm.h>
+#endif
+
 namespace gbm_wrapper {
 
 namespace {
diff --git a/ui/ozone/common/stub_client_native_pixmap_factory.cc b/ui/ozone/common/stub_client_native_pixmap_factory.cc
index f5b33302848a..15be9b464397 100644
--- a/ui/ozone/common/stub_client_native_pixmap_factory.cc
+++ b/ui/ozone/common/stub_client_native_pixmap_factory.cc
@@ -2,6 +2,8 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+#include <memory>
+
 #include "base/macros.h"
 #include "ui/gfx/client_native_pixmap_factory.h"
 
diff --git a/ui/ozone/common/stub_overlay_manager.cc b/ui/ozone/common/stub_overlay_manager.cc
index 5a91846682c5..a905ea833d19 100644
--- a/ui/ozone/common/stub_overlay_manager.cc
+++ b/ui/ozone/common/stub_overlay_manager.cc
@@ -3,6 +3,9 @@
 // found in the LICENSE file.
 
 #include "ui/ozone/common/stub_overlay_manager.h"
+
+#include <memory>
+
 #include "ui/ozone/public/overlay_candidates_ozone.h"
 
 namespace ui {
diff --git a/ui/ozone/common/stub_overlay_manager.h b/ui/ozone/common/stub_overlay_manager.h
index 238a93cf9304..2ef33ef2ccf9 100644
--- a/ui/ozone/common/stub_overlay_manager.h
+++ b/ui/ozone/common/stub_overlay_manager.h
@@ -5,6 +5,8 @@
 #ifndef UI_OZONE_COMMON_STUB_OVERLAY_MANAGER_H_
 #define UI_OZONE_COMMON_STUB_OVERLAY_MANAGER_H_
 
+#include <memory>
+
 #include "base/macros.h"
 #include "ui/ozone/public/overlay_manager_ozone.h"
 
diff --git a/ui/ozone/demo/demo_window.cc b/ui/ozone/demo/demo_window.cc
index f376744e90ea..d21fe30ca30d 100644
--- a/ui/ozone/demo/demo_window.cc
+++ b/ui/ozone/demo/demo_window.cc
@@ -4,6 +4,8 @@
 
 #include "ui/ozone/demo/demo_window.h"
 
+#include <utility>
+
 #include "base/threading/thread_task_runner_handle.h"
 #include "ui/events/event.h"
 #include "ui/events/keycodes/dom/dom_code.h"
diff --git a/ui/ozone/demo/demo_window.h b/ui/ozone/demo/demo_window.h
index a0be6aead252..9987837dca09 100644
--- a/ui/ozone/demo/demo_window.h
+++ b/ui/ozone/demo/demo_window.h
@@ -5,6 +5,8 @@
 #ifndef UI_OZONE_DEMO_DEMO_WINDOW_H_
 #define UI_OZONE_DEMO_DEMO_WINDOW_H_
 
+#include <memory>
+
 #include "base/memory/weak_ptr.h"
 #include "ui/gfx/geometry/rect.h"
 #include "ui/gfx/geometry/size.h"
diff --git a/ui/ozone/demo/gl_renderer.cc b/ui/ozone/demo/gl_renderer.cc
index 61d6a0741f24..44220e4905cb 100644
--- a/ui/ozone/demo/gl_renderer.cc
+++ b/ui/ozone/demo/gl_renderer.cc
@@ -4,6 +4,9 @@
 
 #include "ui/ozone/demo/gl_renderer.h"
 
+#include <memory>
+#include <utility>
+
 #include "base/bind.h"
 #include "base/location.h"
 #include "base/threading/thread_task_runner_handle.h"
diff --git a/ui/ozone/demo/gl_renderer.h b/ui/ozone/demo/gl_renderer.h
index 9837b120eb36..e1e2c2571a36 100644
--- a/ui/ozone/demo/gl_renderer.h
+++ b/ui/ozone/demo/gl_renderer.h
@@ -5,6 +5,8 @@
 #ifndef UI_OZONE_DEMO_GL_RENDERER_H_
 #define UI_OZONE_DEMO_GL_RENDERER_H_
 
+#include <memory>
+
 #include "base/macros.h"
 #include "base/memory/ref_counted.h"
 #include "base/memory/weak_ptr.h"
diff --git a/ui/ozone/demo/ozone_demo.cc b/ui/ozone/demo/ozone_demo.cc
index 11379e5ab057..9867a9b444a1 100644
--- a/ui/ozone/demo/ozone_demo.cc
+++ b/ui/ozone/demo/ozone_demo.cc
@@ -3,6 +3,7 @@
 // found in the LICENSE file.
 
 #include <iostream>
+#include <memory>
 
 #include "base/at_exit.h"
 #include "base/command_line.h"
diff --git a/ui/ozone/demo/renderer_factory.h b/ui/ozone/demo/renderer_factory.h
index a39a3b416c0c..e2fc9a6401bc 100644
--- a/ui/ozone/demo/renderer_factory.h
+++ b/ui/ozone/demo/renderer_factory.h
@@ -5,6 +5,8 @@
 #ifndef UI_OZONE_DEMO_RENDERER_FACTORY_H_
 #define UI_OZONE_DEMO_RENDERER_FACTORY_H_
 
+#include <memory>
+
 #include "ui/gfx/geometry/size.h"
 #include "ui/gfx/native_widget_types.h"
 
diff --git a/ui/ozone/demo/simple_renderer_factory.cc b/ui/ozone/demo/simple_renderer_factory.cc
index 328cfe419e07..b94e0b649419 100644
--- a/ui/ozone/demo/simple_renderer_factory.cc
+++ b/ui/ozone/demo/simple_renderer_factory.cc
@@ -5,6 +5,7 @@
 #include "ui/ozone/demo/simple_renderer_factory.h"
 
 #include <memory>
+#include <utility>
 
 #include "base/command_line.h"
 #include "ui/gl/gl_surface.h"
diff --git a/ui/ozone/demo/simple_renderer_factory.h b/ui/ozone/demo/simple_renderer_factory.h
index 135619e80c3b..98206cd58394 100644
--- a/ui/ozone/demo/simple_renderer_factory.h
+++ b/ui/ozone/demo/simple_renderer_factory.h
@@ -5,6 +5,8 @@
 #ifndef UI_OZONE_DEMO_SIMPLE_RENDERER_FACTORY_H_
 #define UI_OZONE_DEMO_SIMPLE_RENDERER_FACTORY_H_
 
+#include <memory>
+
 #include "gpu/vulkan/buildflags.h"
 #include "ui/ozone/demo/renderer_factory.h"
 #include "ui/ozone/public/ozone_gpu_test_helper.h"
diff --git a/ui/ozone/demo/skia/skia_demo.cc b/ui/ozone/demo/skia/skia_demo.cc
index 2eee7b2afe7e..5d31f13ce560 100644
--- a/ui/ozone/demo/skia/skia_demo.cc
+++ b/ui/ozone/demo/skia/skia_demo.cc
@@ -3,6 +3,7 @@
 // found in the LICENSE file.
 
 #include <iostream>
+#include <memory>
 
 #include "base/at_exit.h"
 #include "base/command_line.h"
diff --git a/ui/ozone/demo/skia/skia_gl_renderer.cc b/ui/ozone/demo/skia/skia_gl_renderer.cc
index 59df60818c19..4298f34bd138 100644
--- a/ui/ozone/demo/skia/skia_gl_renderer.cc
+++ b/ui/ozone/demo/skia/skia_gl_renderer.cc
@@ -4,6 +4,9 @@
 
 #include "ui/ozone/demo/skia/skia_gl_renderer.h"
 
+#include <memory>
+#include <utility>
+
 #include "base/bind.h"
 #include "base/command_line.h"
 #include "base/location.h"
diff --git a/ui/ozone/demo/skia/skia_gl_renderer.h b/ui/ozone/demo/skia/skia_gl_renderer.h
index 7bb6772ed6bb..d4c2b48047be 100644
--- a/ui/ozone/demo/skia/skia_gl_renderer.h
+++ b/ui/ozone/demo/skia/skia_gl_renderer.h
@@ -5,6 +5,8 @@
 #ifndef UI_OZONE_DEMO_SKIA_SKIA_GL_RENDERER_H_
 #define UI_OZONE_DEMO_SKIA_SKIA_GL_RENDERER_H_
 
+#include <memory>
+
 #include "base/containers/queue.h"
 #include "base/macros.h"
 #include "base/memory/ref_counted.h"
diff --git a/ui/ozone/demo/skia/skia_renderer_factory.cc b/ui/ozone/demo/skia/skia_renderer_factory.cc
index 604cc1a6d229..9f2d623e12c0 100644
--- a/ui/ozone/demo/skia/skia_renderer_factory.cc
+++ b/ui/ozone/demo/skia/skia_renderer_factory.cc
@@ -5,6 +5,7 @@
 #include "ui/ozone/demo/skia/skia_renderer_factory.h"
 
 #include <memory>
+#include <utility>
 
 #include "base/command_line.h"
 #include "ui/gl/gl_surface.h"
diff --git a/ui/ozone/demo/skia/skia_renderer_factory.h b/ui/ozone/demo/skia/skia_renderer_factory.h
index dd4068f719ed..98f6be8c04a0 100644
--- a/ui/ozone/demo/skia/skia_renderer_factory.h
+++ b/ui/ozone/demo/skia/skia_renderer_factory.h
@@ -5,6 +5,8 @@
 #ifndef UI_OZONE_DEMO_SKIA_SKIA_RENDERER_FACTORY_H_
 #define UI_OZONE_DEMO_SKIA_SKIA_RENDERER_FACTORY_H_
 
+#include <memory>
+
 #include "ui/gfx/geometry/size.h"
 #include "ui/gfx/native_widget_types.h"
 #include "ui/ozone/demo/renderer_factory.h"
diff --git a/ui/ozone/demo/skia/skia_surfaceless_gl_renderer.cc b/ui/ozone/demo/skia/skia_surfaceless_gl_renderer.cc
index 18f47f390921..1c5a91ca9e03 100644
--- a/ui/ozone/demo/skia/skia_surfaceless_gl_renderer.cc
+++ b/ui/ozone/demo/skia/skia_surfaceless_gl_renderer.cc
@@ -5,6 +5,8 @@
 #include "ui/ozone/demo/skia/skia_surfaceless_gl_renderer.h"
 
 #include <stddef.h>
+#include <memory>
+#include <utility>
 
 #include "base/bind.h"
 #include "base/bind_helpers.h"
diff --git a/ui/ozone/demo/software_renderer.cc b/ui/ozone/demo/software_renderer.cc
index 6523bc23e835..1f7d8ceadac9 100644
--- a/ui/ozone/demo/software_renderer.cc
+++ b/ui/ozone/demo/software_renderer.cc
@@ -4,6 +4,9 @@
 
 #include "ui/ozone/demo/software_renderer.h"
 
+#include <memory>
+#include <utility>
+
 #include "base/bind.h"
 #include "base/trace_event/trace_event.h"
 #include "third_party/skia/include/core/SkCanvas.h"
diff --git a/ui/ozone/demo/surfaceless_gl_renderer.cc b/ui/ozone/demo/surfaceless_gl_renderer.cc
index 0160057cdaab..ffaa6fa98d5e 100644
--- a/ui/ozone/demo/surfaceless_gl_renderer.cc
+++ b/ui/ozone/demo/surfaceless_gl_renderer.cc
@@ -5,6 +5,8 @@
 #include "ui/ozone/demo/surfaceless_gl_renderer.h"
 
 #include <stddef.h>
+#include <memory>
+#include <utility>
 
 #include "base/bind.h"
 #include "base/bind_helpers.h"
diff --git a/ui/ozone/demo/vulkan_overlay_renderer.cc b/ui/ozone/demo/vulkan_overlay_renderer.cc
index a7056fbac492..9f50c6bf0559 100644
--- a/ui/ozone/demo/vulkan_overlay_renderer.cc
+++ b/ui/ozone/demo/vulkan_overlay_renderer.cc
@@ -5,6 +5,8 @@
 #include "ui/ozone/demo/vulkan_overlay_renderer.h"
 
 #include <vulkan/vulkan.h>
+#include <memory>
+#include <utility>
 
 #include "base/bind.h"
 #include "base/location.h"
diff --git a/ui/ozone/demo/vulkan_overlay_renderer.h b/ui/ozone/demo/vulkan_overlay_renderer.h
index 37b751251982..28bdb8d0e591 100644
--- a/ui/ozone/demo/vulkan_overlay_renderer.h
+++ b/ui/ozone/demo/vulkan_overlay_renderer.h
@@ -6,6 +6,7 @@
 #define UI_OZONE_DEMO_VULKAN_OVERLAY_RENDERER_H_
 
 #include <vulkan/vulkan.h>
+#include <memory>
 
 #include "base/macros.h"
 #include "base/memory/ref_counted.h"
diff --git a/ui/ozone/demo/vulkan_renderer.cc b/ui/ozone/demo/vulkan_renderer.cc
index d309099fe943..d19890c89eea 100644
--- a/ui/ozone/demo/vulkan_renderer.cc
+++ b/ui/ozone/demo/vulkan_renderer.cc
@@ -5,6 +5,8 @@
 #include "ui/ozone/demo/vulkan_renderer.h"
 
 #include <vulkan/vulkan.h>
+#include <memory>
+#include <utility>
 
 #include "base/bind.h"
 #include "base/location.h"
diff --git a/ui/ozone/demo/vulkan_renderer.h b/ui/ozone/demo/vulkan_renderer.h
index cd7e6554a9a9..2aaff48b6af5 100644
--- a/ui/ozone/demo/vulkan_renderer.h
+++ b/ui/ozone/demo/vulkan_renderer.h
@@ -6,6 +6,7 @@
 #define UI_OZONE_DEMO_VULKAN_RENDERER_H_
 
 #include <vulkan/vulkan.h>
+#include <memory>
 
 #include "base/macros.h"
 #include "base/memory/ref_counted.h"
diff --git a/ui/ozone/demo/window_manager.cc b/ui/ozone/demo/window_manager.cc
index 219e9072a8fe..ffea5853158e 100644
--- a/ui/ozone/demo/window_manager.cc
+++ b/ui/ozone/demo/window_manager.cc
@@ -4,6 +4,9 @@
 
 #include "ui/ozone/demo/window_manager.h"
 
+#include <memory>
+#include <utility>
+
 #include "base/bind.h"
 #include "base/command_line.h"
 #include "base/threading/thread_task_runner_handle.h"
diff --git a/ui/ozone/demo/window_manager.h b/ui/ozone/demo/window_manager.h
index ae96df0bbba5..a31b01d47bb5 100644
--- a/ui/ozone/demo/window_manager.h
+++ b/ui/ozone/demo/window_manager.h
@@ -5,6 +5,7 @@
 #ifndef UI_OZONE_DEMO_WINDOW_MANAGER_H_
 #define UI_OZONE_DEMO_WINDOW_MANAGER_H_
 
+#include <memory>
 #include <vector>
 
 #include "base/callback.h"
diff --git a/ui/ozone/gl/gl_image_ozone_native_pixmap_unittest.cc b/ui/ozone/gl/gl_image_ozone_native_pixmap_unittest.cc
index 532b040a9105..f66e5e937121 100644
--- a/ui/ozone/gl/gl_image_ozone_native_pixmap_unittest.cc
+++ b/ui/ozone/gl/gl_image_ozone_native_pixmap_unittest.cc
@@ -3,7 +3,6 @@
 // found in the LICENSE file.
 
 #include <stdint.h>
-
 #include <memory>
 
 #include "testing/gtest/include/gtest/gtest.h"
diff --git a/ui/ozone/platform/cast/client_native_pixmap_factory_cast.cc b/ui/ozone/platform/cast/client_native_pixmap_factory_cast.cc
index 60c79d0621f0..36fc30bd2239 100644
--- a/ui/ozone/platform/cast/client_native_pixmap_factory_cast.cc
+++ b/ui/ozone/platform/cast/client_native_pixmap_factory_cast.cc
@@ -4,6 +4,8 @@
 
 #include "ui/ozone/platform/cast/client_native_pixmap_factory_cast.h"
 
+#include <memory>
+
 #include "base/logging.h"
 #include "ui/gfx/buffer_types.h"
 #include "ui/gfx/client_native_pixmap.h"
diff --git a/ui/ozone/platform/cast/gl_ozone_egl_cast.cc b/ui/ozone/platform/cast/gl_ozone_egl_cast.cc
index 6e560e1b606e..446afea24fc2 100644
--- a/ui/ozone/platform/cast/gl_ozone_egl_cast.cc
+++ b/ui/ozone/platform/cast/gl_ozone_egl_cast.cc
@@ -6,7 +6,7 @@
 
 #include <EGL/egl.h>
 #include <dlfcn.h>
-
+#include <memory>
 #include <utility>
 
 #include "base/command_line.h"
diff --git a/ui/ozone/platform/cast/gl_ozone_egl_cast.h b/ui/ozone/platform/cast/gl_ozone_egl_cast.h
index 6a4a1c7bfb48..ace0cbc94abd 100644
--- a/ui/ozone/platform/cast/gl_ozone_egl_cast.h
+++ b/ui/ozone/platform/cast/gl_ozone_egl_cast.h
@@ -6,7 +6,6 @@
 #define UI_OZONE_PLATFORM_CAST_GL_OZONE_EGL_CAST_H_
 
 #include <stdint.h>
-
 #include <memory>
 
 #include "base/macros.h"
diff --git a/ui/ozone/platform/cast/gl_surface_cast.cc b/ui/ozone/platform/cast/gl_surface_cast.cc
index 3378fd0f9c2c..d76bfacb1f8d 100644
--- a/ui/ozone/platform/cast/gl_surface_cast.cc
+++ b/ui/ozone/platform/cast/gl_surface_cast.cc
@@ -4,6 +4,7 @@
 
 #include "ui/ozone/platform/cast/gl_surface_cast.h"
 
+#include <memory>
 #include <string>
 #include <utility>
 
diff --git a/ui/ozone/platform/cast/gl_surface_cast.h b/ui/ozone/platform/cast/gl_surface_cast.h
index e8d88a6f34db..e294802478fb 100644
--- a/ui/ozone/platform/cast/gl_surface_cast.h
+++ b/ui/ozone/platform/cast/gl_surface_cast.h
@@ -5,6 +5,7 @@
 #ifndef UI_OZONE_PLATFORM_CAST_GL_SURFACE_CAST_H_
 #define UI_OZONE_PLATFORM_CAST_GL_SURFACE_CAST_H_
 
+#include <memory>
 #include <vector>
 
 #include "base/macros.h"
diff --git a/ui/ozone/platform/cast/overlay_manager_cast.cc b/ui/ozone/platform/cast/overlay_manager_cast.cc
index 66a210abc79d..32bbae781a1c 100644
--- a/ui/ozone/platform/cast/overlay_manager_cast.cc
+++ b/ui/ozone/platform/cast/overlay_manager_cast.cc
@@ -4,6 +4,8 @@
 
 #include "ui/ozone/platform/cast/overlay_manager_cast.h"
 
+#include <memory>
+
 #include "ui/ozone/public/overlay_candidates_ozone.h"
 
 namespace ui {
diff --git a/ui/ozone/platform/cast/ozone_platform_cast.cc b/ui/ozone/platform/cast/ozone_platform_cast.cc
index 594d0e63ccbc..5a6d6352c4cd 100644
--- a/ui/ozone/platform/cast/ozone_platform_cast.cc
+++ b/ui/ozone/platform/cast/ozone_platform_cast.cc
@@ -4,6 +4,7 @@
 
 #include "ui/ozone/platform/cast/ozone_platform_cast.h"
 
+#include <memory>
 #include <utility>
 
 #include "base/command_line.h"
diff --git a/ui/ozone/platform/cast/surface_factory_cast.cc b/ui/ozone/platform/cast/surface_factory_cast.cc
index 3bca9130e947..107e11a64027 100644
--- a/ui/ozone/platform/cast/surface_factory_cast.cc
+++ b/ui/ozone/platform/cast/surface_factory_cast.cc
@@ -4,6 +4,7 @@
 
 #include "ui/ozone/platform/cast/surface_factory_cast.h"
 
+#include <memory>
 #include <utility>
 
 #include "base/macros.h"
diff --git a/ui/ozone/platform/drm/common/display_types.h b/ui/ozone/platform/drm/common/display_types.h
index da8b5c075337..c7cb3ef562cd 100644
--- a/ui/ozone/platform/drm/common/display_types.h
+++ b/ui/ozone/platform/drm/common/display_types.h
@@ -5,6 +5,8 @@
 #ifndef UI_OZONE_PLATFORM_DRM_COMMON_DISPLAY_TYPES_H_
 #define UI_OZONE_PLATFORM_DRM_COMMON_DISPLAY_TYPES_H_
 
+#include <memory>
+
 namespace display {
 class DisplaySnapshot;
 }  // namespace display
@@ -16,4 +18,4 @@ using MovableDisplaySnapshots =
 
 }  // namespace ui
 
-#endif  // UI_OZONE_PLATFORM_DRM_COMMON_DISPLAY_TYPES_H_
\ No newline at end of file
+#endif  // UI_OZONE_PLATFORM_DRM_COMMON_DISPLAY_TYPES_H_
diff --git a/ui/ozone/platform/drm/common/drm_overlay_manager.cc b/ui/ozone/platform/drm/common/drm_overlay_manager.cc
index 1881a979c845..380283495e6b 100644
--- a/ui/ozone/platform/drm/common/drm_overlay_manager.cc
+++ b/ui/ozone/platform/drm/common/drm_overlay_manager.cc
@@ -5,6 +5,7 @@
 #include "ui/ozone/platform/drm/common/drm_overlay_manager.h"
 
 #include <algorithm>
+#include <memory>
 #include <utility>
 
 #include "base/trace_event/trace_event.h"
diff --git a/ui/ozone/platform/drm/common/drm_util.cc b/ui/ozone/platform/drm/common/drm_util.cc
index 76a6caf83c51..dd9d0f74d528 100644
--- a/ui/ozone/platform/drm/common/drm_util.cc
+++ b/ui/ozone/platform/drm/common/drm_util.cc
@@ -11,6 +11,7 @@
 #include <xf86drm.h>
 #include <xf86drmMode.h>
 #include <algorithm>
+#include <memory>
 #include <utility>
 
 #include "base/containers/flat_map.h"
diff --git a/ui/ozone/platform/drm/common/drm_util.h b/ui/ozone/platform/drm/common/drm_util.h
index e70c23a8d9ef..c46941ffe296 100644
--- a/ui/ozone/platform/drm/common/drm_util.h
+++ b/ui/ozone/platform/drm/common/drm_util.h
@@ -8,6 +8,7 @@
 #include <stddef.h>
 
 #include <memory>
+#include <utility>
 #include <vector>
 
 #include "base/files/file_path.h"
diff --git a/ui/ozone/platform/drm/gpu/crtc_controller.cc b/ui/ozone/platform/drm/gpu/crtc_controller.cc
index aea5c918d4db..d23d0dd6c223 100644
--- a/ui/ozone/platform/drm/gpu/crtc_controller.cc
+++ b/ui/ozone/platform/drm/gpu/crtc_controller.cc
@@ -4,6 +4,8 @@
 
 #include "ui/ozone/platform/drm/gpu/crtc_controller.h"
 
+#include <memory>
+
 #include "base/logging.h"
 #include "base/time/time.h"
 #include "ui/gfx/presentation_feedback.h"
diff --git a/ui/ozone/platform/drm/gpu/drm_device.cc b/ui/ozone/platform/drm/gpu/drm_device.cc
index 313bf2dfdb8f..11b699eae63a 100644
--- a/ui/ozone/platform/drm/gpu/drm_device.cc
+++ b/ui/ozone/platform/drm/gpu/drm_device.cc
@@ -9,6 +9,7 @@
 #include <unistd.h>
 #include <xf86drm.h>
 #include <xf86drmMode.h>
+#include <memory>
 #include <utility>
 
 #include "base/bind.h"
diff --git a/ui/ozone/platform/drm/gpu/drm_device.h b/ui/ozone/platform/drm/gpu/drm_device.h
index 1e62f1891c9a..d5ef13b1b6fd 100644
--- a/ui/ozone/platform/drm/gpu/drm_device.h
+++ b/ui/ozone/platform/drm/gpu/drm_device.h
@@ -7,7 +7,7 @@
 
 #include <stddef.h>
 #include <stdint.h>
-
+#include <memory>
 #include <vector>
 
 #include "base/callback.h"
diff --git a/ui/ozone/platform/drm/gpu/drm_device_manager.cc b/ui/ozone/platform/drm/gpu/drm_device_manager.cc
index 6b3214aad248..5d879fa98533 100644
--- a/ui/ozone/platform/drm/gpu/drm_device_manager.cc
+++ b/ui/ozone/platform/drm/gpu/drm_device_manager.cc
@@ -4,6 +4,7 @@
 
 #include "ui/ozone/platform/drm/gpu/drm_device_manager.h"
 
+#include <memory>
 #include <utility>
 
 #include "base/file_descriptor_posix.h"
diff --git a/ui/ozone/platform/drm/gpu/drm_display.cc b/ui/ozone/platform/drm/gpu/drm_display.cc
index f800442687d3..f5bf25f8c4f0 100644
--- a/ui/ozone/platform/drm/gpu/drm_display.cc
+++ b/ui/ozone/platform/drm/gpu/drm_display.cc
@@ -5,6 +5,7 @@
 #include "ui/ozone/platform/drm/gpu/drm_display.h"
 
 #include <xf86drmMode.h>
+#include <memory>
 
 #include "base/stl_util.h"
 #include "ui/display/types/display_snapshot.h"
diff --git a/ui/ozone/platform/drm/gpu/drm_display.h b/ui/ozone/platform/drm/gpu/drm_display.h
index 16f2dae49961..d0f591140433 100644
--- a/ui/ozone/platform/drm/gpu/drm_display.h
+++ b/ui/ozone/platform/drm/gpu/drm_display.h
@@ -7,7 +7,7 @@
 
 #include <stddef.h>
 #include <stdint.h>
-
+#include <memory>
 #include <vector>
 
 #include "base/macros.h"
diff --git a/ui/ozone/platform/drm/gpu/drm_framebuffer.cc b/ui/ozone/platform/drm/gpu/drm_framebuffer.cc
index b632257a5736..e348da188a79 100644
--- a/ui/ozone/platform/drm/gpu/drm_framebuffer.cc
+++ b/ui/ozone/platform/drm/gpu/drm_framebuffer.cc
@@ -4,6 +4,8 @@
 
 #include "ui/ozone/platform/drm/gpu/drm_framebuffer.h"
 
+#include <utility>
+
 #include "ui/ozone/common/linux/drm_util_linux.h"
 #include "ui/ozone/common/linux/gbm_buffer.h"
 #include "ui/ozone/platform/drm/common/drm_util.h"
diff --git a/ui/ozone/platform/drm/gpu/drm_gpu_display_manager.cc b/ui/ozone/platform/drm/gpu/drm_gpu_display_manager.cc
index 232ae7694242..8ce481254c55 100644
--- a/ui/ozone/platform/drm/gpu/drm_gpu_display_manager.cc
+++ b/ui/ozone/platform/drm/gpu/drm_gpu_display_manager.cc
@@ -5,6 +5,8 @@
 #include "ui/ozone/platform/drm/gpu/drm_gpu_display_manager.h"
 
 #include <stddef.h>
+#include <memory>
+#include <utility>
 
 #include "ui/display/types/display_mode.h"
 #include "ui/display/types/display_snapshot.h"
diff --git a/ui/ozone/platform/drm/gpu/drm_gpu_display_manager.h b/ui/ozone/platform/drm/gpu/drm_gpu_display_manager.h
index 4a7669faddf9..40d71deaf4b2 100644
--- a/ui/ozone/platform/drm/gpu/drm_gpu_display_manager.h
+++ b/ui/ozone/platform/drm/gpu/drm_gpu_display_manager.h
@@ -6,7 +6,6 @@
 #define UI_OZONE_PLATFORM_DRM_GPU_DRM_GPU_DISPLAY_MANAGER_H_
 
 #include <stdint.h>
-
 #include <memory>
 #include <vector>
 
diff --git a/ui/ozone/platform/drm/gpu/drm_overlay_plane.cc b/ui/ozone/platform/drm/gpu/drm_overlay_plane.cc
index 87fa16d7d1e6..72a00b7dc1a9 100644
--- a/ui/ozone/platform/drm/gpu/drm_overlay_plane.cc
+++ b/ui/ozone/platform/drm/gpu/drm_overlay_plane.cc
@@ -5,6 +5,8 @@
 #include "ui/ozone/platform/drm/gpu/drm_overlay_plane.h"
 
 #include <stddef.h>
+#include <memory>
+#include <utility>
 
 #include "ui/gfx/gpu_fence.h"
 #include "ui/ozone/platform/drm/gpu/drm_framebuffer.h"
diff --git a/ui/ozone/platform/drm/gpu/drm_overlay_plane.h b/ui/ozone/platform/drm/gpu/drm_overlay_plane.h
index e0dcf10a7267..1e25e55a68aa 100644
--- a/ui/ozone/platform/drm/gpu/drm_overlay_plane.h
+++ b/ui/ozone/platform/drm/gpu/drm_overlay_plane.h
@@ -5,6 +5,7 @@
 #ifndef UI_OZONE_PLATFORM_DRM_GPU_DRM_OVERLAY_PLANE_H_
 #define UI_OZONE_PLATFORM_DRM_GPU_DRM_OVERLAY_PLANE_H_
 
+#include <memory>
 #include <vector>
 
 #include "base/bind.h"
diff --git a/ui/ozone/platform/drm/gpu/drm_overlay_validator.cc b/ui/ozone/platform/drm/gpu/drm_overlay_validator.cc
index ea0947a19ef9..d0fe45480a6e 100644
--- a/ui/ozone/platform/drm/gpu/drm_overlay_validator.cc
+++ b/ui/ozone/platform/drm/gpu/drm_overlay_validator.cc
@@ -5,6 +5,8 @@
 #include "ui/ozone/platform/drm/gpu/drm_overlay_validator.h"
 
 #include <drm_fourcc.h>
+#include <memory>
+#include <utility>
 
 #include "base/files/platform_file.h"
 #include "ui/gfx/geometry/size_conversions.h"
diff --git a/ui/ozone/platform/drm/gpu/drm_overlay_validator_unittest.cc b/ui/ozone/platform/drm/gpu/drm_overlay_validator_unittest.cc
index 7aa42855b804..2f78b21a92af 100644
--- a/ui/ozone/platform/drm/gpu/drm_overlay_validator_unittest.cc
+++ b/ui/ozone/platform/drm/gpu/drm_overlay_validator_unittest.cc
@@ -5,7 +5,6 @@
 #include "ui/ozone/platform/drm/gpu/drm_overlay_validator.h"
 
 #include <drm_fourcc.h>
-
 #include <memory>
 #include <utility>
 
diff --git a/ui/ozone/platform/drm/gpu/drm_thread.cc b/ui/ozone/platform/drm/gpu/drm_thread.cc
index 42cea80685be..3b53a0d383b4 100644
--- a/ui/ozone/platform/drm/gpu/drm_thread.cc
+++ b/ui/ozone/platform/drm/gpu/drm_thread.cc
@@ -5,6 +5,7 @@
 #include "ui/ozone/platform/drm/gpu/drm_thread.h"
 
 #include <gbm.h>
+#include <memory>
 #include <utility>
 
 #include "base/bind.h"
diff --git a/ui/ozone/platform/drm/gpu/drm_thread.h b/ui/ozone/platform/drm/gpu/drm_thread.h
index 47f09370b152..fb9b070c0805 100644
--- a/ui/ozone/platform/drm/gpu/drm_thread.h
+++ b/ui/ozone/platform/drm/gpu/drm_thread.h
@@ -6,7 +6,6 @@
 #define UI_OZONE_PLATFORM_DRM_GPU_DRM_THREAD_H_
 
 #include <stdint.h>
-
 #include <memory>
 
 #include "base/files/file.h"
diff --git a/ui/ozone/platform/drm/gpu/drm_thread_message_proxy.cc b/ui/ozone/platform/drm/gpu/drm_thread_message_proxy.cc
index cf65c7bedbd0..ece302bad369 100644
--- a/ui/ozone/platform/drm/gpu/drm_thread_message_proxy.cc
+++ b/ui/ozone/platform/drm/gpu/drm_thread_message_proxy.cc
@@ -4,6 +4,8 @@
 
 #include "ui/ozone/platform/drm/gpu/drm_thread_message_proxy.h"
 
+#include <utility>
+
 #include "base/bind.h"
 #include "base/task_runner_util.h"
 #include "base/threading/thread_task_runner_handle.h"
diff --git a/ui/ozone/platform/drm/gpu/drm_thread_proxy.cc b/ui/ozone/platform/drm/gpu/drm_thread_proxy.cc
index 4c97c66af23e..f26113081958 100644
--- a/ui/ozone/platform/drm/gpu/drm_thread_proxy.cc
+++ b/ui/ozone/platform/drm/gpu/drm_thread_proxy.cc
@@ -4,6 +4,9 @@
 
 #include "ui/ozone/platform/drm/gpu/drm_thread_proxy.h"
 
+#include <memory>
+#include <utility>
+
 #include "base/bind.h"
 #include "ui/ozone/platform/drm/gpu/drm_thread_message_proxy.h"
 #include "ui/ozone/platform/drm/gpu/drm_window_proxy.h"
diff --git a/ui/ozone/platform/drm/gpu/drm_window.cc b/ui/ozone/platform/drm/gpu/drm_window.cc
index c2aadb98bf1b..99688b533e0d 100644
--- a/ui/ozone/platform/drm/gpu/drm_window.cc
+++ b/ui/ozone/platform/drm/gpu/drm_window.cc
@@ -6,6 +6,7 @@
 
 #include <stddef.h>
 #include <stdint.h>
+#include <utility>
 
 #include "base/macros.h"
 #include "base/time/time.h"
diff --git a/ui/ozone/platform/drm/gpu/drm_window.h b/ui/ozone/platform/drm/gpu/drm_window.h
index ad34c452b9a5..f0fbdb1ea2c2 100644
--- a/ui/ozone/platform/drm/gpu/drm_window.h
+++ b/ui/ozone/platform/drm/gpu/drm_window.h
@@ -5,6 +5,7 @@
 #ifndef UI_OZONE_PLATFORM_DRM_GPU_DRM_WINDOW_H_
 #define UI_OZONE_PLATFORM_DRM_GPU_DRM_WINDOW_H_
 
+#include <memory>
 #include <vector>
 
 #include "base/macros.h"
diff --git a/ui/ozone/platform/drm/gpu/drm_window_proxy.cc b/ui/ozone/platform/drm/gpu/drm_window_proxy.cc
index 8f06b91fe2db..45ed5b6d95d2 100644
--- a/ui/ozone/platform/drm/gpu/drm_window_proxy.cc
+++ b/ui/ozone/platform/drm/gpu/drm_window_proxy.cc
@@ -4,6 +4,8 @@
 
 #include "ui/ozone/platform/drm/gpu/drm_window_proxy.h"
 
+#include <utility>
+
 #include "base/bind.h"
 #include "base/command_line.h"
 #include "ui/gfx/gpu_fence.h"
diff --git a/ui/ozone/platform/drm/gpu/drm_window_unittest.cc b/ui/ozone/platform/drm/gpu/drm_window_unittest.cc
index 94233bc51b9a..b2a01444c3c8 100644
--- a/ui/ozone/platform/drm/gpu/drm_window_unittest.cc
+++ b/ui/ozone/platform/drm/gpu/drm_window_unittest.cc
@@ -6,7 +6,6 @@
 
 #include <drm_fourcc.h>
 #include <stdint.h>
-
 #include <memory>
 #include <utility>
 #include <vector>
diff --git a/ui/ozone/platform/drm/gpu/gbm_overlay_surface.cc b/ui/ozone/platform/drm/gpu/gbm_overlay_surface.cc
index d8f6f5729739..fecf60e5b182 100644
--- a/ui/ozone/platform/drm/gpu/gbm_overlay_surface.cc
+++ b/ui/ozone/platform/drm/gpu/gbm_overlay_surface.cc
@@ -5,6 +5,8 @@
 #include "ui/ozone/platform/drm/gpu/gbm_overlay_surface.h"
 
 #include <unistd.h>
+#include <memory>
+#include <utility>
 
 #include "base/bind.h"
 #include "base/bind_helpers.h"
diff --git a/ui/ozone/platform/drm/gpu/gbm_pixmap.cc b/ui/ozone/platform/drm/gpu/gbm_pixmap.cc
index 17463810e67a..2d80356fcab5 100644
--- a/ui/ozone/platform/drm/gpu/gbm_pixmap.cc
+++ b/ui/ozone/platform/drm/gpu/gbm_pixmap.cc
@@ -5,6 +5,8 @@
 #include "ui/ozone/platform/drm/gpu/gbm_pixmap.h"
 
 #include <gbm.h>
+#include <memory>
+#include <utility>
 
 #include "base/logging.h"
 #include "ui/gfx/gpu_fence.h"
diff --git a/ui/ozone/platform/drm/gpu/gbm_pixmap.h b/ui/ozone/platform/drm/gpu/gbm_pixmap.h
index 6646759c69c6..60c856c430ab 100644
--- a/ui/ozone/platform/drm/gpu/gbm_pixmap.h
+++ b/ui/ozone/platform/drm/gpu/gbm_pixmap.h
@@ -5,6 +5,7 @@
 #ifndef UI_OZONE_PLATFORM_DRM_GPU_GBM_BUFFER_H_
 #define UI_OZONE_PLATFORM_DRM_GPU_GBM_BUFFER_H_
 
+#include <memory>
 #include <vector>
 
 #include "ui/gfx/native_pixmap.h"
diff --git a/ui/ozone/platform/drm/gpu/gbm_surface_factory.cc b/ui/ozone/platform/drm/gpu/gbm_surface_factory.cc
index 55a2d8fe2a6f..1e4fa9ccdb61 100644
--- a/ui/ozone/platform/drm/gpu/gbm_surface_factory.cc
+++ b/ui/ozone/platform/drm/gpu/gbm_surface_factory.cc
@@ -5,7 +5,7 @@
 #include "ui/ozone/platform/drm/gpu/gbm_surface_factory.h"
 
 #include <gbm.h>
-
+#include <memory>
 #include <utility>
 
 #include "base/files/file_path.h"
diff --git a/ui/ozone/platform/drm/gpu/gbm_surface_factory.h b/ui/ozone/platform/drm/gpu/gbm_surface_factory.h
index 028a1687a5d9..5d4c822a4134 100644
--- a/ui/ozone/platform/drm/gpu/gbm_surface_factory.h
+++ b/ui/ozone/platform/drm/gpu/gbm_surface_factory.h
@@ -6,7 +6,6 @@
 #define UI_OZONE_PLATFORM_DRM_GPU_GBM_SURFACE_FACTORY_H_
 
 #include <stdint.h>
-
 #include <map>
 #include <memory>
 #include <vector>
diff --git a/ui/ozone/platform/drm/gpu/gbm_surfaceless.cc b/ui/ozone/platform/drm/gpu/gbm_surfaceless.cc
index 66d000a6c2d5..67b984329f82 100644
--- a/ui/ozone/platform/drm/gpu/gbm_surfaceless.cc
+++ b/ui/ozone/platform/drm/gpu/gbm_surfaceless.cc
@@ -4,6 +4,7 @@
 
 #include "ui/ozone/platform/drm/gpu/gbm_surfaceless.h"
 
+#include <memory>
 #include <utility>
 
 #include "base/bind.h"
diff --git a/ui/ozone/platform/drm/gpu/hardware_display_controller.cc b/ui/ozone/platform/drm/gpu/hardware_display_controller.cc
index 8dbb15ca84ef..828f2e44f25f 100644
--- a/ui/ozone/platform/drm/gpu/hardware_display_controller.cc
+++ b/ui/ozone/platform/drm/gpu/hardware_display_controller.cc
@@ -7,6 +7,7 @@
 #include <drm.h>
 #include <string.h>
 #include <xf86drm.h>
+#include <memory>
 #include <utility>
 
 #include "base/bind.h"
diff --git a/ui/ozone/platform/drm/gpu/hardware_display_controller.h b/ui/ozone/platform/drm/gpu/hardware_display_controller.h
index 1031f12cb096..7e6cb78769af 100644
--- a/ui/ozone/platform/drm/gpu/hardware_display_controller.h
+++ b/ui/ozone/platform/drm/gpu/hardware_display_controller.h
@@ -8,7 +8,6 @@
 #include <stddef.h>
 #include <stdint.h>
 #include <xf86drmMode.h>
-
 #include <map>
 #include <memory>
 #include <unordered_map>
diff --git a/ui/ozone/platform/drm/gpu/hardware_display_controller_unittest.cc b/ui/ozone/platform/drm/gpu/hardware_display_controller_unittest.cc
index 28c210a8222a..d5ba180f39af 100644
--- a/ui/ozone/platform/drm/gpu/hardware_display_controller_unittest.cc
+++ b/ui/ozone/platform/drm/gpu/hardware_display_controller_unittest.cc
@@ -5,6 +5,8 @@
 #include <drm_fourcc.h>
 #include <stddef.h>
 #include <stdint.h>
+#include <memory>
+#include <utility>
 
 #include "base/bind.h"
 #include "base/macros.h"
diff --git a/ui/ozone/platform/drm/gpu/hardware_display_plane_manager.cc b/ui/ozone/platform/drm/gpu/hardware_display_plane_manager.cc
index bb8799b0626b..35eb919baffd 100644
--- a/ui/ozone/platform/drm/gpu/hardware_display_plane_manager.cc
+++ b/ui/ozone/platform/drm/gpu/hardware_display_plane_manager.cc
@@ -5,8 +5,8 @@
 #include "ui/ozone/platform/drm/gpu/hardware_display_plane_manager.h"
 
 #include <drm_fourcc.h>
-
 #include <algorithm>
+#include <memory>
 #include <set>
 #include <utility>
 
diff --git a/ui/ozone/platform/drm/gpu/hardware_display_plane_manager.h b/ui/ozone/platform/drm/gpu/hardware_display_plane_manager.h
index 45cd06ce454c..f68e178054dd 100644
--- a/ui/ozone/platform/drm/gpu/hardware_display_plane_manager.h
+++ b/ui/ozone/platform/drm/gpu/hardware_display_plane_manager.h
@@ -8,6 +8,7 @@
 #include <stddef.h>
 #include <stdint.h>
 #include <xf86drmMode.h>
+#include <memory>
 #include <vector>
 
 #include "base/macros.h"
diff --git a/ui/ozone/platform/drm/gpu/hardware_display_plane_manager_atomic.cc b/ui/ozone/platform/drm/gpu/hardware_display_plane_manager_atomic.cc
index c1868d0f9139..ae429e679676 100644
--- a/ui/ozone/platform/drm/gpu/hardware_display_plane_manager_atomic.cc
+++ b/ui/ozone/platform/drm/gpu/hardware_display_plane_manager_atomic.cc
@@ -7,6 +7,8 @@
 #include <sync/sync.h>
 #include <xf86drm.h>
 #include <xf86drmMode.h>
+#include <memory>
+#include <utility>
 
 #include "base/bind.h"
 #include "base/files/platform_file.h"
diff --git a/ui/ozone/platform/drm/gpu/hardware_display_plane_manager_atomic.h b/ui/ozone/platform/drm/gpu/hardware_display_plane_manager_atomic.h
index b3196146234b..c259643ef5aa 100644
--- a/ui/ozone/platform/drm/gpu/hardware_display_plane_manager_atomic.h
+++ b/ui/ozone/platform/drm/gpu/hardware_display_plane_manager_atomic.h
@@ -6,6 +6,7 @@
 #define UI_OZONE_PLATFORM_DRM_GPU_HARDWARE_DISPLAY_PLANE_MANAGER_ATOMIC_H_
 
 #include <stdint.h>
+#include <memory>
 
 #include "base/macros.h"
 #include "ui/ozone/platform/drm/gpu/hardware_display_plane_manager.h"
diff --git a/ui/ozone/platform/drm/gpu/hardware_display_plane_manager_legacy.cc b/ui/ozone/platform/drm/gpu/hardware_display_plane_manager_legacy.cc
index 498aec547fc8..136f991c7ace 100644
--- a/ui/ozone/platform/drm/gpu/hardware_display_plane_manager_legacy.cc
+++ b/ui/ozone/platform/drm/gpu/hardware_display_plane_manager_legacy.cc
@@ -6,6 +6,8 @@
 
 #include <errno.h>
 #include <sync/sync.h>
+#include <memory>
+#include <utility>
 
 #include "base/bind.h"
 #include "base/posix/eintr_wrapper.h"
diff --git a/ui/ozone/platform/drm/gpu/hardware_display_plane_manager_legacy.h b/ui/ozone/platform/drm/gpu/hardware_display_plane_manager_legacy.h
index a40413ff2d04..3843560218ec 100644
--- a/ui/ozone/platform/drm/gpu/hardware_display_plane_manager_legacy.h
+++ b/ui/ozone/platform/drm/gpu/hardware_display_plane_manager_legacy.h
@@ -6,6 +6,7 @@
 #define UI_OZONE_PLATFORM_DRM_GPU_HARDWARE_DISPLAY_PLANE_MANAGER_LEGACY_H_
 
 #include <stdint.h>
+#include <memory>
 
 #include "base/macros.h"
 #include "ui/ozone/platform/drm/gpu/hardware_display_plane_manager.h"
diff --git a/ui/ozone/platform/drm/gpu/hardware_display_plane_manager_unittest.cc b/ui/ozone/platform/drm/gpu/hardware_display_plane_manager_unittest.cc
index 8c2a3d2daa96..cfd093053fea 100644
--- a/ui/ozone/platform/drm/gpu/hardware_display_plane_manager_unittest.cc
+++ b/ui/ozone/platform/drm/gpu/hardware_display_plane_manager_unittest.cc
@@ -5,8 +5,8 @@
 #include <drm_fourcc.h>
 #include <stdint.h>
 #include <unistd.h>
-
 #include <memory>
+#include <utility>
 
 #include "base/bind.h"
 #include "base/files/file_util.h"
diff --git a/ui/ozone/platform/drm/gpu/mock_drm_device.cc b/ui/ozone/platform/drm/gpu/mock_drm_device.cc
index e9dfffbfb41e..38cc4f3fe047 100644
--- a/ui/ozone/platform/drm/gpu/mock_drm_device.cc
+++ b/ui/ozone/platform/drm/gpu/mock_drm_device.cc
@@ -5,6 +5,8 @@
 #include "ui/ozone/platform/drm/gpu/mock_drm_device.h"
 
 #include <xf86drm.h>
+#include <memory>
+#include <utility>
 
 #include "base/logging.h"
 #include "base/stl_util.h"
diff --git a/ui/ozone/platform/drm/gpu/mock_drm_device.h b/ui/ozone/platform/drm/gpu/mock_drm_device.h
index 64279da0cfd4..ab824d51302c 100644
--- a/ui/ozone/platform/drm/gpu/mock_drm_device.h
+++ b/ui/ozone/platform/drm/gpu/mock_drm_device.h
@@ -8,8 +8,8 @@
 #include <drm_mode.h>
 #include <stddef.h>
 #include <stdint.h>
-
 #include <map>
+#include <memory>
 #include <set>
 #include <vector>
 
diff --git a/ui/ozone/platform/drm/gpu/mock_gbm_device.cc b/ui/ozone/platform/drm/gpu/mock_gbm_device.cc
index 713478e2c0cf..20d6f1f038cf 100644
--- a/ui/ozone/platform/drm/gpu/mock_gbm_device.cc
+++ b/ui/ozone/platform/drm/gpu/mock_gbm_device.cc
@@ -6,6 +6,8 @@
 
 #include <drm_fourcc.h>
 #include <xf86drm.h>
+#include <memory>
+#include <utility>
 
 #include "base/logging.h"
 #include "base/numerics/safe_math.h"
diff --git a/ui/ozone/platform/drm/gpu/page_flip_request.cc b/ui/ozone/platform/drm/gpu/page_flip_request.cc
index 7f37668fcc96..b863742bdc1f 100644
--- a/ui/ozone/platform/drm/gpu/page_flip_request.cc
+++ b/ui/ozone/platform/drm/gpu/page_flip_request.cc
@@ -4,6 +4,8 @@
 
 #include "ui/ozone/platform/drm/gpu/page_flip_request.h"
 
+#include <utility>
+
 #include "base/bind.h"
 #include "ui/gfx/presentation_feedback.h"
 
diff --git a/ui/ozone/platform/drm/gpu/proxy_helpers.cc b/ui/ozone/platform/drm/gpu/proxy_helpers.cc
index 72dca98a2f5a..603b3bc15817 100644
--- a/ui/ozone/platform/drm/gpu/proxy_helpers.cc
+++ b/ui/ozone/platform/drm/gpu/proxy_helpers.cc
@@ -4,6 +4,8 @@
 
 #include "ui/ozone/platform/drm/gpu/proxy_helpers.h"
 
+#include <utility>
+
 #include "base/synchronization/waitable_event.h"
 
 namespace ui {
diff --git a/ui/ozone/platform/drm/gpu/proxy_helpers.h b/ui/ozone/platform/drm/gpu/proxy_helpers.h
index 8f7c3260525e..b0d8a5244c32 100644
--- a/ui/ozone/platform/drm/gpu/proxy_helpers.h
+++ b/ui/ozone/platform/drm/gpu/proxy_helpers.h
@@ -5,6 +5,8 @@
 #ifndef UI_OZONE_PLATFORM_DRM_GPU_PROXY_HELPERS_H_
 #define UI_OZONE_PLATFORM_DRM_GPU_PROXY_HELPERS_H_
 
+#include <utility>
+
 #include "base/bind.h"
 #include "base/location.h"
 #include "base/memory/ref_counted.h"
diff --git a/ui/ozone/platform/drm/gpu/proxy_helpers_unittest.cc b/ui/ozone/platform/drm/gpu/proxy_helpers_unittest.cc
index 9831443f4d6c..b96b3908a332 100644
--- a/ui/ozone/platform/drm/gpu/proxy_helpers_unittest.cc
+++ b/ui/ozone/platform/drm/gpu/proxy_helpers_unittest.cc
@@ -4,6 +4,9 @@
 
 #include "ui/ozone/platform/drm/gpu/proxy_helpers.h"
 
+#include <memory>
+#include <utility>
+
 #include "base/bind.h"
 #include "base/message_loop/message_loop.h"
 #include "base/run_loop.h"
diff --git a/ui/ozone/platform/drm/gpu/screen_manager.cc b/ui/ozone/platform/drm/gpu/screen_manager.cc
index 5fe1773665d6..894283b504b1 100644
--- a/ui/ozone/platform/drm/gpu/screen_manager.cc
+++ b/ui/ozone/platform/drm/gpu/screen_manager.cc
@@ -5,7 +5,7 @@
 #include "ui/ozone/platform/drm/gpu/screen_manager.h"
 
 #include <xf86drmMode.h>
-
+#include <memory>
 #include <utility>
 
 #include "base/files/platform_file.h"
diff --git a/ui/ozone/platform/drm/gpu/screen_manager.h b/ui/ozone/platform/drm/gpu/screen_manager.h
index e5c2760dc5bc..9b1aeb2ea4ac 100644
--- a/ui/ozone/platform/drm/gpu/screen_manager.h
+++ b/ui/ozone/platform/drm/gpu/screen_manager.h
@@ -6,7 +6,7 @@
 #define UI_OZONE_PLATFORM_DRM_GPU_SCREEN_MANAGER_H_
 
 #include <stdint.h>
-
+#include <memory>
 #include <unordered_map>
 
 #include "base/macros.h"
diff --git a/ui/ozone/platform/drm/gpu/screen_manager_unittest.cc b/ui/ozone/platform/drm/gpu/screen_manager_unittest.cc
index 4deec5b31657..c4d137dba189 100644
--- a/ui/ozone/platform/drm/gpu/screen_manager_unittest.cc
+++ b/ui/ozone/platform/drm/gpu/screen_manager_unittest.cc
@@ -5,7 +5,7 @@
 #include <drm_fourcc.h>
 #include <stddef.h>
 #include <stdint.h>
-
+#include <memory>
 #include <utility>
 
 #include "base/bind_helpers.h"
diff --git a/ui/ozone/platform/drm/gpu/vulkan_implementation_gbm.cc b/ui/ozone/platform/drm/gpu/vulkan_implementation_gbm.cc
index 21e13af55c9b..6564a8854438 100644
--- a/ui/ozone/platform/drm/gpu/vulkan_implementation_gbm.cc
+++ b/ui/ozone/platform/drm/gpu/vulkan_implementation_gbm.cc
@@ -4,6 +4,8 @@
 
 #include "ui/ozone/platform/drm/gpu/vulkan_implementation_gbm.h"
 
+#include <memory>
+
 #include "base/files/file_path.h"
 #include "base/native_library.h"
 #include "gpu/vulkan/vulkan_function_pointers.h"
diff --git a/ui/ozone/platform/drm/host/drm_cursor.cc b/ui/ozone/platform/drm/host/drm_cursor.cc
index 3b8e789c5018..8e420a59c347 100644
--- a/ui/ozone/platform/drm/host/drm_cursor.cc
+++ b/ui/ozone/platform/drm/host/drm_cursor.cc
@@ -4,6 +4,9 @@
 
 #include "ui/ozone/platform/drm/host/drm_cursor.h"
 
+#include <memory>
+#include <utility>
+
 #include "base/trace_event/trace_event.h"
 #include "ui/gfx/geometry/point_conversions.h"
 #include "ui/ozone/platform/drm/host/drm_window_host.h"
diff --git a/ui/ozone/platform/drm/host/drm_device_connector.cc b/ui/ozone/platform/drm/host/drm_device_connector.cc
index 2da65e7f4b77..d207dbe14f9e 100644
--- a/ui/ozone/platform/drm/host/drm_device_connector.cc
+++ b/ui/ozone/platform/drm/host/drm_device_connector.cc
@@ -4,6 +4,8 @@
 
 #include "ui/ozone/platform/drm/host/drm_device_connector.h"
 
+#include <utility>
+
 #include "base/bind.h"
 #include "mojo/public/cpp/bindings/interface_request.h"
 #include "mojo/public/cpp/system/message_pipe.h"
diff --git a/ui/ozone/platform/drm/host/drm_display_host.cc b/ui/ozone/platform/drm/host/drm_display_host.cc
index 78271ca2af00..a0bcdbcc06c5 100644
--- a/ui/ozone/platform/drm/host/drm_display_host.cc
+++ b/ui/ozone/platform/drm/host/drm_display_host.cc
@@ -4,6 +4,9 @@
 
 #include "ui/ozone/platform/drm/host/drm_display_host.h"
 
+#include <memory>
+#include <utility>
+
 #include "base/bind.h"
 #include "base/location.h"
 #include "base/threading/thread_task_runner_handle.h"
diff --git a/ui/ozone/platform/drm/host/drm_display_host_manager.cc b/ui/ozone/platform/drm/host/drm_display_host_manager.cc
index 6aaee531f7ba..5e7db1093fec 100644
--- a/ui/ozone/platform/drm/host/drm_display_host_manager.cc
+++ b/ui/ozone/platform/drm/host/drm_display_host_manager.cc
@@ -7,7 +7,7 @@
 #include <fcntl.h>
 #include <stddef.h>
 #include <xf86drm.h>
-
+#include <memory>
 #include <utility>
 
 #include "base/bind.h"
diff --git a/ui/ozone/platform/drm/host/drm_display_host_manager.h b/ui/ozone/platform/drm/host/drm_display_host_manager.h
index 4e91b193a815..2ed263a239a0 100644
--- a/ui/ozone/platform/drm/host/drm_display_host_manager.h
+++ b/ui/ozone/platform/drm/host/drm_display_host_manager.h
@@ -6,7 +6,6 @@
 #define UI_OZONE_PLATFORM_DRM_HOST_DRM_DISPLAY_HOST_MANAGER_H_
 
 #include <stdint.h>
-
 #include <map>
 #include <memory>
 
diff --git a/ui/ozone/platform/drm/host/drm_gpu_platform_support_host.cc b/ui/ozone/platform/drm/host/drm_gpu_platform_support_host.cc
index e4677fb81ada..4e353fb4d48a 100644
--- a/ui/ozone/platform/drm/host/drm_gpu_platform_support_host.cc
+++ b/ui/ozone/platform/drm/host/drm_gpu_platform_support_host.cc
@@ -5,6 +5,7 @@
 #include "ui/ozone/platform/drm/host/drm_gpu_platform_support_host.h"
 
 #include <stddef.h>
+#include <utility>
 
 #include "base/bind.h"
 #include "base/command_line.h"
diff --git a/ui/ozone/platform/drm/host/drm_native_display_delegate.cc b/ui/ozone/platform/drm/host/drm_native_display_delegate.cc
index 61d507f1a50d..54ecf769e5b5 100644
--- a/ui/ozone/platform/drm/host/drm_native_display_delegate.cc
+++ b/ui/ozone/platform/drm/host/drm_native_display_delegate.cc
@@ -4,6 +4,8 @@
 
 #include "ui/ozone/platform/drm/host/drm_native_display_delegate.h"
 
+#include <utility>
+
 #include "ui/display/types/display_snapshot.h"
 #include "ui/display/types/native_display_observer.h"
 #include "ui/ozone/platform/drm/host/drm_display_host.h"
diff --git a/ui/ozone/platform/drm/host/host_cursor_proxy.cc b/ui/ozone/platform/drm/host/host_cursor_proxy.cc
index 2a6e34247afb..766980d2de2a 100644
--- a/ui/ozone/platform/drm/host/host_cursor_proxy.cc
+++ b/ui/ozone/platform/drm/host/host_cursor_proxy.cc
@@ -4,6 +4,8 @@
 
 #include "ui/ozone/platform/drm/host/host_cursor_proxy.h"
 
+#include <utility>
+
 #include "services/service_manager/public/cpp/connector.h"
 #include "services/ws/public/mojom/constants.mojom.h"
 #include "ui/ozone/public/gpu_platform_support_host.h"
diff --git a/ui/ozone/platform/drm/host/host_drm_device.cc b/ui/ozone/platform/drm/host/host_drm_device.cc
index a82b7e9cecde..4ff514f94064 100644
--- a/ui/ozone/platform/drm/host/host_drm_device.cc
+++ b/ui/ozone/platform/drm/host/host_drm_device.cc
@@ -4,6 +4,9 @@
 
 #include "ui/ozone/platform/drm/host/host_drm_device.h"
 
+#include <memory>
+#include <utility>
+
 #include "base/bind.h"
 #include "base/run_loop.h"
 #include "base/single_thread_task_runner.h"
diff --git a/ui/ozone/platform/drm/host/host_drm_device.h b/ui/ozone/platform/drm/host/host_drm_device.h
index 30f61c101748..7c5187791ffb 100644
--- a/ui/ozone/platform/drm/host/host_drm_device.h
+++ b/ui/ozone/platform/drm/host/host_drm_device.h
@@ -5,6 +5,8 @@
 #ifndef UI_OZONE_PLATFORM_DRM_HOST_HOST_DRM_DEVICE_H_
 #define UI_OZONE_PLATFORM_DRM_HOST_HOST_DRM_DEVICE_H_
 
+#include <memory>
+
 #include "base/callback.h"
 #include "base/macros.h"
 #include "base/memory/weak_ptr.h"
diff --git a/ui/ozone/platform/drm/ozone_platform_gbm.cc b/ui/ozone/platform/drm/ozone_platform_gbm.cc
index bb078a5d51ee..f26217bab9e8 100644
--- a/ui/ozone/platform/drm/ozone_platform_gbm.cc
+++ b/ui/ozone/platform/drm/ozone_platform_gbm.cc
@@ -7,7 +7,6 @@
 #include <gbm.h>
 #include <stdlib.h>
 #include <xf86drm.h>
-
 #include <memory>
 #include <utility>
 
diff --git a/ui/ozone/platform/headless/headless_native_display_delegate.cc b/ui/ozone/platform/headless/headless_native_display_delegate.cc
index a4ca1d32bce6..adb117538225 100644
--- a/ui/ozone/platform/headless/headless_native_display_delegate.cc
+++ b/ui/ozone/platform/headless/headless_native_display_delegate.cc
@@ -4,6 +4,9 @@
 
 #include "ui/ozone/platform/headless/headless_native_display_delegate.h"
 
+#include <memory>
+#include <utility>
+
 #include "ui/display/types/display_snapshot.h"
 #include "ui/display/types/native_display_observer.h"
 
diff --git a/ui/ozone/platform/headless/headless_native_display_delegate.h b/ui/ozone/platform/headless/headless_native_display_delegate.h
index 37ac5ba69e1e..1cd9402a5dee 100644
--- a/ui/ozone/platform/headless/headless_native_display_delegate.h
+++ b/ui/ozone/platform/headless/headless_native_display_delegate.h
@@ -5,6 +5,8 @@
 #ifndef UI_OZONE_PLATFORM_HEADLESS_HEADLESS_NATIVE_DISPLAY_DELEGATE_H_
 #define UI_OZONE_PLATFORM_HEADLESS_HEADLESS_NATIVE_DISPLAY_DELEGATE_H_
 
+#include <memory>
+
 #include "base/observer_list.h"
 #include "ui/display/types/native_display_delegate.h"
 
diff --git a/ui/ozone/platform/headless/headless_surface_factory.cc b/ui/ozone/platform/headless/headless_surface_factory.cc
index d1ef15f8a0b5..3c3a0b75232e 100644
--- a/ui/ozone/platform/headless/headless_surface_factory.cc
+++ b/ui/ozone/platform/headless/headless_surface_factory.cc
@@ -4,6 +4,8 @@
 
 #include "ui/ozone/platform/headless/headless_surface_factory.h"
 
+#include <memory>
+
 #include "base/bind.h"
 #include "base/files/file_util.h"
 #include "base/location.h"
diff --git a/ui/ozone/platform/headless/ozone_platform_headless.cc b/ui/ozone/platform/headless/ozone_platform_headless.cc
index 3e22233d1ec6..85279dd72771 100644
--- a/ui/ozone/platform/headless/ozone_platform_headless.cc
+++ b/ui/ozone/platform/headless/ozone_platform_headless.cc
@@ -4,6 +4,8 @@
 
 #include "ui/ozone/platform/headless/ozone_platform_headless.h"
 
+#include <memory>
+
 #include "base/command_line.h"
 #include "base/files/file_path.h"
 #include "base/macros.h"
diff --git a/ui/ozone/platform/magma/ozone_platform_magma.cc b/ui/ozone/platform/magma/ozone_platform_magma.cc
index f5467a436abd..40c6b827635d 100644
--- a/ui/ozone/platform/magma/ozone_platform_magma.cc
+++ b/ui/ozone/platform/magma/ozone_platform_magma.cc
@@ -4,6 +4,8 @@
 
 #include "ui/ozone/platform/magma/ozone_platform_magma.h"
 
+#include <memory>
+
 #include "base/command_line.h"
 #include "base/files/file_path.h"
 #include "base/macros.h"
diff --git a/ui/ozone/platform/scenic/scenic_gpu_host.cc b/ui/ozone/platform/scenic/scenic_gpu_host.cc
index 758de960fccc..9e9925e35471 100644
--- a/ui/ozone/platform/scenic/scenic_gpu_host.cc
+++ b/ui/ozone/platform/scenic/scenic_gpu_host.cc
@@ -5,6 +5,7 @@
 #include "ui/ozone/platform/scenic/scenic_gpu_host.h"
 
 #include <inttypes.h>
+#include <utility>
 
 #include "base/bind.h"
 #include "base/callback.h"
diff --git a/ui/ozone/platform/scenic/scenic_surface_factory.cc b/ui/ozone/platform/scenic/scenic_surface_factory.cc
index c89627279d35..637aa44e076b 100644
--- a/ui/ozone/platform/scenic/scenic_surface_factory.cc
+++ b/ui/ozone/platform/scenic/scenic_surface_factory.cc
@@ -5,6 +5,7 @@
 #include "ui/ozone/platform/scenic/scenic_surface_factory.h"
 
 #include <lib/zx/event.h>
+#include <memory>
 
 #include "base/bind.h"
 #include "base/fuchsia/fuchsia_logging.h"
diff --git a/ui/ozone/platform/scenic/scenic_window_canvas.cc b/ui/ozone/platform/scenic/scenic_window_canvas.cc
index 155914838982..9f8d43e3dace 100644
--- a/ui/ozone/platform/scenic/scenic_window_canvas.cc
+++ b/ui/ozone/platform/scenic/scenic_window_canvas.cc
@@ -4,6 +4,8 @@
 
 #include "ui/ozone/platform/scenic/scenic_window_canvas.h"
 
+#include <memory>
+
 #include "base/fuchsia/fuchsia_logging.h"
 #include "base/memory/read_only_shared_memory_region.h"
 #include "base/memory/writable_shared_memory_region.h"
diff --git a/ui/ozone/platform/scenic/scenic_window_canvas.h b/ui/ozone/platform/scenic/scenic_window_canvas.h
index f385a9650ce1..237e621a105c 100644
--- a/ui/ozone/platform/scenic/scenic_window_canvas.h
+++ b/ui/ozone/platform/scenic/scenic_window_canvas.h
@@ -6,6 +6,7 @@
 #define UI_OZONE_PLATFORM_SCENIC_SCENIC_WINDOW_CANVAS_H_
 
 #include <lib/ui/scenic/cpp/resources.h>
+#include <memory>
 
 #include "base/macros.h"
 #include "base/memory/shared_memory_mapping.h"
diff --git a/ui/ozone/platform/scenic/scenic_window_manager.cc b/ui/ozone/platform/scenic/scenic_window_manager.cc
index aeabc0eeccbd..da55cee93eef 100644
--- a/ui/ozone/platform/scenic/scenic_window_manager.cc
+++ b/ui/ozone/platform/scenic/scenic_window_manager.cc
@@ -4,6 +4,8 @@
 
 #include "ui/ozone/platform/scenic/scenic_window_manager.h"
 
+#include <memory>
+
 #include "base/fuchsia/fuchsia_logging.h"
 #include "base/fuchsia/service_directory_client.h"
 #include "ui/ozone/platform/scenic/ozone_platform_scenic.h"
diff --git a/ui/ozone/platform/scenic/vulkan_implementation_scenic.cc b/ui/ozone/platform/scenic/vulkan_implementation_scenic.cc
index 72184009d44d..e82f0b34fa5d 100644
--- a/ui/ozone/platform/scenic/vulkan_implementation_scenic.cc
+++ b/ui/ozone/platform/scenic/vulkan_implementation_scenic.cc
@@ -8,6 +8,7 @@
 #include <lib/ui/scenic/cpp/session.h>
 #include <lib/zx/channel.h>
 #include <vulkan/vulkan.h>
+#include <memory>
 
 #include "base/bind_helpers.h"
 #include "base/files/file_path.h"
diff --git a/ui/ozone/platform/wayland/gl_surface_wayland.cc b/ui/ozone/platform/wayland/gl_surface_wayland.cc
index 5827383eefee..1c65f6d3eb76 100644
--- a/ui/ozone/platform/wayland/gl_surface_wayland.cc
+++ b/ui/ozone/platform/wayland/gl_surface_wayland.cc
@@ -5,7 +5,7 @@
 #include "ui/ozone/platform/wayland/gl_surface_wayland.h"
 
 #include <wayland-egl.h>
-
+#include <memory>
 #include <utility>
 
 #include "third_party/khronos/EGL/egl.h"
diff --git a/ui/ozone/platform/wayland/gpu/gbm_pixmap_wayland.cc b/ui/ozone/platform/wayland/gpu/gbm_pixmap_wayland.cc
index c2c1ea086610..ca6ccf6152e1 100644
--- a/ui/ozone/platform/wayland/gpu/gbm_pixmap_wayland.cc
+++ b/ui/ozone/platform/wayland/gpu/gbm_pixmap_wayland.cc
@@ -7,6 +7,7 @@
 #include <drm_fourcc.h>
 #include <gbm.h>
 #include <xf86drmMode.h>
+#include <memory>
 
 #include "base/files/platform_file.h"
 #include "base/logging.h"
diff --git a/ui/ozone/platform/wayland/gpu/gbm_pixmap_wayland.h b/ui/ozone/platform/wayland/gpu/gbm_pixmap_wayland.h
index 3c191ceea95a..3f43538be354 100644
--- a/ui/ozone/platform/wayland/gpu/gbm_pixmap_wayland.h
+++ b/ui/ozone/platform/wayland/gpu/gbm_pixmap_wayland.h
@@ -5,6 +5,7 @@
 #ifndef UI_OZONE_PLATFORM_WAYLAND_GPU_GBM_PIXMAP_WAYLAND_H_
 #define UI_OZONE_PLATFORM_WAYLAND_GPU_GBM_PIXMAP_WAYLAND_H_
 
+#include <memory>
 #include <vector>
 
 #include "base/files/scoped_file.h"
diff --git a/ui/ozone/platform/wayland/gpu/gbm_surfaceless_wayland.cc b/ui/ozone/platform/wayland/gpu/gbm_surfaceless_wayland.cc
index 175f0b691d60..49d29381fda0 100644
--- a/ui/ozone/platform/wayland/gpu/gbm_surfaceless_wayland.cc
+++ b/ui/ozone/platform/wayland/gpu/gbm_surfaceless_wayland.cc
@@ -4,6 +4,8 @@
 
 #include "ui/ozone/platform/wayland/gpu/gbm_surfaceless_wayland.h"
 
+#include <memory>
+
 #include "base/bind.h"
 #include "base/task/post_task.h"
 #include "base/trace_event/trace_event.h"
diff --git a/ui/ozone/platform/wayland/gpu/gbm_surfaceless_wayland.h b/ui/ozone/platform/wayland/gpu/gbm_surfaceless_wayland.h
index 6870debbc761..f86dd4b0c9c6 100644
--- a/ui/ozone/platform/wayland/gpu/gbm_surfaceless_wayland.h
+++ b/ui/ozone/platform/wayland/gpu/gbm_surfaceless_wayland.h
@@ -5,6 +5,8 @@
 #ifndef UI_OZONE_PLATFORM_WAYLAND_GPU_GBM_SURFACELESS_WAYLAND_H_
 #define UI_OZONE_PLATFORM_WAYLAND_GPU_GBM_SURFACELESS_WAYLAND_H_
 
+#include <memory>
+
 #include "base/macros.h"
 #include "base/memory/weak_ptr.h"
 #include "ui/gfx/native_widget_types.h"
diff --git a/ui/ozone/platform/wayland/gpu/wayland_canvas_surface.cc b/ui/ozone/platform/wayland/gpu/wayland_canvas_surface.cc
index a281d5bd578a..a8fb25a25b5f 100644
--- a/ui/ozone/platform/wayland/gpu/wayland_canvas_surface.cc
+++ b/ui/ozone/platform/wayland/gpu/wayland_canvas_surface.cc
@@ -4,6 +4,7 @@
 
 #include "ui/ozone/platform/wayland/gpu/wayland_canvas_surface.h"
 
+#include <memory>
 #include <utility>
 
 #include "base/files/scoped_file.h"
diff --git a/ui/ozone/platform/wayland/gpu/wayland_connection_proxy.h b/ui/ozone/platform/wayland/gpu/wayland_connection_proxy.h
index 27909874410d..13a8d0437d26 100644
--- a/ui/ozone/platform/wayland/gpu/wayland_connection_proxy.h
+++ b/ui/ozone/platform/wayland/gpu/wayland_connection_proxy.h
@@ -5,6 +5,8 @@
 #ifndef UI_OZONE_PLATFORM_WAYLAND_GPU_WAYLAND_CONNECTION_PROXY_H_
 #define UI_OZONE_PLATFORM_WAYLAND_GPU_WAYLAND_CONNECTION_PROXY_H_
 
+#include <memory>
+
 #include "base/macros.h"
 #include "base/threading/sequenced_task_runner_handle.h"
 #include "base/threading/thread_checker.h"
diff --git a/ui/ozone/platform/wayland/test/test_wayland_server_thread.cc b/ui/ozone/platform/wayland/test/test_wayland_server_thread.cc
index e92e077ce027..99e8a3ef121f 100644
--- a/ui/ozone/platform/wayland/test/test_wayland_server_thread.cc
+++ b/ui/ozone/platform/wayland/test/test_wayland_server_thread.cc
@@ -7,6 +7,7 @@
 #include <stdlib.h>
 #include <sys/socket.h>
 #include <wayland-server.h>
+#include <memory>
 
 #include "base/bind.h"
 #include "base/files/file_util.h"
diff --git a/ui/ozone/platform/wayland/test/test_wayland_server_thread.h b/ui/ozone/platform/wayland/test/test_wayland_server_thread.h
index 24c00bf578d3..fddd426db690 100644
--- a/ui/ozone/platform/wayland/test/test_wayland_server_thread.h
+++ b/ui/ozone/platform/wayland/test/test_wayland_server_thread.h
@@ -7,7 +7,6 @@
 
 #include <memory>
 #include <vector>
-
 #include <wayland-server-core.h>
 
 #include "base/message_loop/message_pump_libevent.h"
diff --git a/ui/ozone/platform/wayland/wayland_buffer_fuzzer.cc b/ui/ozone/platform/wayland/wayland_buffer_fuzzer.cc
index f10f93366941..117208ccee5d 100644
--- a/ui/ozone/platform/wayland/wayland_buffer_fuzzer.cc
+++ b/ui/ozone/platform/wayland/wayland_buffer_fuzzer.cc
@@ -8,7 +8,6 @@
 #include <drm_fourcc.h>
 #include <stddef.h>
 #include <stdint.h>
-
 #include <memory>
 #include <vector>
 
diff --git a/ui/ozone/platform/wayland/wayland_buffer_manager.cc b/ui/ozone/platform/wayland/wayland_buffer_manager.cc
index 862989e10cfa..7b0bf9aac1f2 100644
--- a/ui/ozone/platform/wayland/wayland_buffer_manager.cc
+++ b/ui/ozone/platform/wayland/wayland_buffer_manager.cc
@@ -7,6 +7,7 @@
 #include <drm_fourcc.h>
 #include <linux-dmabuf-unstable-v1-client-protocol.h>
 #include <presentation-time-client-protocol.h>
+#include <memory>
 
 #include "base/trace_event/trace_event.h"
 #include "ui/ozone/common/linux/drm_util_linux.h"
diff --git a/ui/ozone/platform/wayland/wayland_buffer_manager.h b/ui/ozone/platform/wayland/wayland_buffer_manager.h
index 50e44bb059a2..0c33947e132a 100644
--- a/ui/ozone/platform/wayland/wayland_buffer_manager.h
+++ b/ui/ozone/platform/wayland/wayland_buffer_manager.h
@@ -6,6 +6,7 @@
 #define UI_OZONE_PLATFORM_WAYLAND_WAYLAND_BUFFER_MANAGER_H_
 
 #include <map>
+#include <memory>
 #include <vector>
 
 #include "base/containers/flat_map.h"
diff --git a/ui/ozone/platform/wayland/wayland_connection.cc b/ui/ozone/platform/wayland/wayland_connection.cc
index bbcb18eb5898..ca36a1be9ecc 100644
--- a/ui/ozone/platform/wayland/wayland_connection.cc
+++ b/ui/ozone/platform/wayland/wayland_connection.cc
@@ -6,6 +6,7 @@
 
 #include <xdg-shell-unstable-v5-client-protocol.h>
 #include <xdg-shell-unstable-v6-client-protocol.h>
+#include <memory>
 
 #include <algorithm>
 #include <utility>
diff --git a/ui/ozone/platform/wayland/wayland_cursor.cc b/ui/ozone/platform/wayland/wayland_cursor.cc
index 725e70cfcf40..deb864165580 100644
--- a/ui/ozone/platform/wayland/wayland_cursor.cc
+++ b/ui/ozone/platform/wayland/wayland_cursor.cc
@@ -4,6 +4,9 @@
 
 #include "ui/ozone/platform/wayland/wayland_cursor.h"
 
+#include <memory>
+#include <vector>
+
 #include "base/memory/shared_memory.h"
 #include "third_party/skia/include/core/SkBitmap.h"
 #include "ui/gfx/skia_util.h"
diff --git a/ui/ozone/platform/wayland/wayland_data_device.cc b/ui/ozone/platform/wayland/wayland_data_device.cc
index ab5df3f4d501..dfe42f2bef37 100644
--- a/ui/ozone/platform/wayland/wayland_data_device.cc
+++ b/ui/ozone/platform/wayland/wayland_data_device.cc
@@ -4,6 +4,7 @@
 
 #include "ui/ozone/platform/wayland/wayland_data_device.h"
 
+#include <memory>
 #include <utility>
 
 #include "base/bind.h"
diff --git a/ui/ozone/platform/wayland/wayland_data_device_unittest.cc b/ui/ozone/platform/wayland/wayland_data_device_unittest.cc
index b09b8cb10ffa..50c4d7e0b456 100644
--- a/ui/ozone/platform/wayland/wayland_data_device_unittest.cc
+++ b/ui/ozone/platform/wayland/wayland_data_device_unittest.cc
@@ -4,6 +4,8 @@
 
 #include <wayland-server.h>
 
+#include <memory>
+
 #include "base/bind.h"
 #include "testing/gtest/include/gtest/gtest.h"
 #include "ui/base/dragdrop/drag_drop_types.h"
diff --git a/ui/ozone/platform/wayland/wayland_input_method_context_factory.cc b/ui/ozone/platform/wayland/wayland_input_method_context_factory.cc
index 2b8349adef06..7537f27eb66c 100644
--- a/ui/ozone/platform/wayland/wayland_input_method_context_factory.cc
+++ b/ui/ozone/platform/wayland/wayland_input_method_context_factory.cc
@@ -4,6 +4,8 @@
 
 #include "ui/ozone/platform/wayland/wayland_input_method_context_factory.h"
 
+#include <memory>
+
 #include "base/bind.h"
 #include "ui/ozone/platform/wayland/wayland_connection.h"
 #include "ui/ozone/platform/wayland/wayland_input_method_context.h"
diff --git a/ui/ozone/platform/wayland/wayland_input_method_context_factory.h b/ui/ozone/platform/wayland/wayland_input_method_context_factory.h
index 413f3a4a500e..23bd6950f0b3 100644
--- a/ui/ozone/platform/wayland/wayland_input_method_context_factory.h
+++ b/ui/ozone/platform/wayland/wayland_input_method_context_factory.h
@@ -5,6 +5,8 @@
 #ifndef UI_OZONE_PLATFORM_WAYLAND_WAYLAND_INPUT_METHOD_CONTEXT_FACTORY_H_
 #define UI_OZONE_PLATFORM_WAYLAND_WAYLAND_INPUT_METHOD_CONTEXT_FACTORY_H_
 
+#include <memory>
+
 #include "base/macros.h"
 #include "ui/base/ime/linux/linux_input_method_context_factory.h"
 
diff --git a/ui/ozone/platform/wayland/wayland_input_method_context_unittest.cc b/ui/ozone/platform/wayland/wayland_input_method_context_unittest.cc
index 08bf4b736943..83ca9f2bbb51 100644
--- a/ui/ozone/platform/wayland/wayland_input_method_context_unittest.cc
+++ b/ui/ozone/platform/wayland/wayland_input_method_context_unittest.cc
@@ -4,6 +4,7 @@
 
 #include <text-input-unstable-v1-server-protocol.h>
 #include <wayland-server.h>
+#include <memory>
 
 #include "mojo/public/cpp/bindings/binding.h"
 #include "testing/gmock/include/gmock/gmock.h"
diff --git a/ui/ozone/platform/wayland/wayland_keyboard_unittest.cc b/ui/ozone/platform/wayland/wayland_keyboard_unittest.cc
index 7d85608ba1e0..3e7abb00d95a 100644
--- a/ui/ozone/platform/wayland/wayland_keyboard_unittest.cc
+++ b/ui/ozone/platform/wayland/wayland_keyboard_unittest.cc
@@ -4,6 +4,7 @@
 
 #include <linux/input.h>
 #include <wayland-server.h>
+#include <memory>
 
 #include "base/test/test_mock_time_task_runner.h"
 #include "base/timer/timer.h"
diff --git a/ui/ozone/platform/wayland/wayland_object.h b/ui/ozone/platform/wayland/wayland_object.h
index f7f7fd3264f5..cdae701b4786 100644
--- a/ui/ozone/platform/wayland/wayland_object.h
+++ b/ui/ozone/platform/wayland/wayland_object.h
@@ -6,7 +6,6 @@
 #define UI_OZONE_PLATFORM_WAYLAND_WAYLAND_OBJECT_H_
 
 #include <wayland-client-core.h>
-
 #include <memory>
 
 struct wl_buffer;
diff --git a/ui/ozone/platform/wayland/wayland_output_manager.cc b/ui/ozone/platform/wayland/wayland_output_manager.cc
index 992d0c309333..a7f6e4471fb2 100644
--- a/ui/ozone/platform/wayland/wayland_output_manager.cc
+++ b/ui/ozone/platform/wayland/wayland_output_manager.cc
@@ -4,6 +4,8 @@
 
 #include "ui/ozone/platform/wayland/wayland_output_manager.h"
 
+#include <memory>
+
 #include "ui/ozone/platform/wayland/wayland_connection.h"
 #include "ui/ozone/platform/wayland/wayland_output.h"
 
diff --git a/ui/ozone/platform/wayland/wayland_pointer.h b/ui/ozone/platform/wayland/wayland_pointer.h
index 109d08db30a9..070e0b6286a3 100644
--- a/ui/ozone/platform/wayland/wayland_pointer.h
+++ b/ui/ozone/platform/wayland/wayland_pointer.h
@@ -5,6 +5,8 @@
 #ifndef UI_OZONE_PLATFORM_WAYLAND_WAYLAND_POINTER_H_
 #define UI_OZONE_PLATFORM_WAYLAND_WAYLAND_POINTER_H_
 
+#include <memory>
+
 #include "base/macros.h"
 #include "base/memory/weak_ptr.h"
 #include "ui/events/ozone/evdev/event_dispatch_callback.h"
diff --git a/ui/ozone/platform/wayland/wayland_pointer_unittest.cc b/ui/ozone/platform/wayland/wayland_pointer_unittest.cc
index 49dcfe77d11e..3b3aa896c730 100644
--- a/ui/ozone/platform/wayland/wayland_pointer_unittest.cc
+++ b/ui/ozone/platform/wayland/wayland_pointer_unittest.cc
@@ -4,6 +4,7 @@
 
 #include <linux/input.h>
 #include <wayland-server.h>
+#include <memory>
 
 #include "testing/gmock/include/gmock/gmock.h"
 #include "testing/gtest/include/gtest/gtest.h"
diff --git a/ui/ozone/platform/wayland/wayland_screen_unittest.cc b/ui/ozone/platform/wayland/wayland_screen_unittest.cc
index a6513be611d9..e081590a6b07 100644
--- a/ui/ozone/platform/wayland/wayland_screen_unittest.cc
+++ b/ui/ozone/platform/wayland/wayland_screen_unittest.cc
@@ -3,7 +3,6 @@
 // found in the LICENSE file.
 
 #include <memory>
-
 #include <wayland-server.h>
 
 #include "testing/gtest/include/gtest/gtest.h"
diff --git a/ui/ozone/platform/wayland/wayland_surface_factory.h b/ui/ozone/platform/wayland/wayland_surface_factory.h
index 2f5ab3a92f17..be0bf798e990 100644
--- a/ui/ozone/platform/wayland/wayland_surface_factory.h
+++ b/ui/ozone/platform/wayland/wayland_surface_factory.h
@@ -5,6 +5,8 @@
 #ifndef UI_OZONE_PLATFORM_WAYLAND_WAYLAND_SURFACE_FACTORY_H_
 #define UI_OZONE_PLATFORM_WAYLAND_WAYLAND_SURFACE_FACTORY_H_
 
+#include <memory>
+
 #include "base/macros.h"
 #include "base/memory/ref_counted.h"
 #include "base/single_thread_task_runner.h"
diff --git a/ui/ozone/platform/wayland/wayland_surface_factory_unittest.cc b/ui/ozone/platform/wayland/wayland_surface_factory_unittest.cc
index 87861f4649d3..f040c6652521 100644
--- a/ui/ozone/platform/wayland/wayland_surface_factory_unittest.cc
+++ b/ui/ozone/platform/wayland/wayland_surface_factory_unittest.cc
@@ -2,6 +2,7 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+#include <memory>
 #include <utility>
 
 #include "base/run_loop.h"
diff --git a/ui/ozone/platform/wayland/wayland_test.h b/ui/ozone/platform/wayland/wayland_test.h
index 625d7e60a941..2c83f37b90c1 100644
--- a/ui/ozone/platform/wayland/wayland_test.h
+++ b/ui/ozone/platform/wayland/wayland_test.h
@@ -5,6 +5,8 @@
 #ifndef UI_OZONE_PLATFORM_WAYLAND_WAYLAND_TEST_H_
 #define UI_OZONE_PLATFORM_WAYLAND_WAYLAND_TEST_H_
 
+#include <memory>
+
 #include "base/message_loop/message_loop.h"
 #include "base/test/scoped_task_environment.h"
 #include "testing/gtest/include/gtest/gtest.h"
diff --git a/ui/ozone/platform/wayland/wayland_touch_unittest.cc b/ui/ozone/platform/wayland/wayland_touch_unittest.cc
index 831dd903cb19..9a64f9c477ec 100644
--- a/ui/ozone/platform/wayland/wayland_touch_unittest.cc
+++ b/ui/ozone/platform/wayland/wayland_touch_unittest.cc
@@ -4,6 +4,7 @@
 
 #include <linux/input.h>
 #include <wayland-server.h>
+#include <memory>
 
 #include "testing/gmock/include/gmock/gmock.h"
 #include "testing/gtest/include/gtest/gtest.h"
diff --git a/ui/ozone/platform/wayland/wayland_window.cc b/ui/ozone/platform/wayland/wayland_window.cc
index 7b658951a403..f0ae1a0b1a80 100644
--- a/ui/ozone/platform/wayland/wayland_window.cc
+++ b/ui/ozone/platform/wayland/wayland_window.cc
@@ -4,9 +4,8 @@
 
 #include "ui/ozone/platform/wayland/wayland_window.h"
 
-#include <memory>
-
 #include <wayland-client.h>
+#include <memory>
 
 #include "base/bind.h"
 #include "ui/base/cursor/ozone/bitmap_cursor_factory_ozone.h"
diff --git a/ui/ozone/platform/wayland/wayland_window.h b/ui/ozone/platform/wayland/wayland_window.h
index 4c7271d19d94..94c3a60c0e96 100644
--- a/ui/ozone/platform/wayland/wayland_window.h
+++ b/ui/ozone/platform/wayland/wayland_window.h
@@ -5,6 +5,7 @@
 #ifndef UI_OZONE_PLATFORM_WAYLAND_WAYLAND_WINDOW_H_
 #define UI_OZONE_PLATFORM_WAYLAND_WAYLAND_WINDOW_H_
 
+#include <memory>
 #include <set>
 #include <vector>
 
diff --git a/ui/ozone/platform/wayland/xdg_popup_wrapper_v6.cc b/ui/ozone/platform/wayland/xdg_popup_wrapper_v6.cc
index 38b274f5d7bd..866133eeba86 100644
--- a/ui/ozone/platform/wayland/xdg_popup_wrapper_v6.cc
+++ b/ui/ozone/platform/wayland/xdg_popup_wrapper_v6.cc
@@ -5,6 +5,7 @@
 #include "ui/ozone/platform/wayland/xdg_popup_wrapper_v6.h"
 
 #include <xdg-shell-unstable-v6-client-protocol.h>
+#include <memory>
 
 #include "ui/events/event_constants.h"
 #include "ui/gfx/geometry/rect.h"
diff --git a/ui/ozone/platform/wayland/xdg_popup_wrapper_v6.h b/ui/ozone/platform/wayland/xdg_popup_wrapper_v6.h
index 4e43d2dd6a4b..cd56eeffd9ca 100644
--- a/ui/ozone/platform/wayland/xdg_popup_wrapper_v6.h
+++ b/ui/ozone/platform/wayland/xdg_popup_wrapper_v6.h
@@ -5,6 +5,8 @@
 #ifndef UI_OZONE_PLATFORM_WAYLAND_XDG_POPUP_WRAPPER_V6_H_
 #define UI_OZONE_PLATFORM_WAYLAND_XDG_POPUP_WRAPPER_V6_H_
 
+#include <memory>
+
 #include "ui/ozone/platform/wayland/xdg_popup_wrapper.h"
 
 namespace ui {
diff --git a/ui/ozone/platform/windows/ozone_platform_windows.cc b/ui/ozone/platform/windows/ozone_platform_windows.cc
index 83469246e349..7b0a598217b1 100644
--- a/ui/ozone/platform/windows/ozone_platform_windows.cc
+++ b/ui/ozone/platform/windows/ozone_platform_windows.cc
@@ -4,6 +4,8 @@
 
 #include "ui/ozone/platform/windows/ozone_platform_windows.h"
 
+#include <memory>
+
 #include "base/command_line.h"
 #include "base/files/file_path.h"
 #include "base/macros.h"
diff --git a/ui/ozone/platform/x11/x11_screen_ozone.h b/ui/ozone/platform/x11/x11_screen_ozone.h
index cd16a1c909cc..5acf82a2eab9 100644
--- a/ui/ozone/platform/x11/x11_screen_ozone.h
+++ b/ui/ozone/platform/x11/x11_screen_ozone.h
@@ -5,6 +5,7 @@
 #ifndef UI_OZONE_PLATFORM_X11_X11_SCREEN_OZONE_H_
 #define UI_OZONE_PLATFORM_X11_X11_SCREEN_OZONE_H_
 
+#include <memory>
 #include <vector>
 
 #include "base/macros.h"
diff --git a/ui/ozone/platform/x11/x11_screen_ozone_unittest.cc b/ui/ozone/platform/x11/x11_screen_ozone_unittest.cc
index 6086f084a464..250da19ce8f4 100644
--- a/ui/ozone/platform/x11/x11_screen_ozone_unittest.cc
+++ b/ui/ozone/platform/x11/x11_screen_ozone_unittest.cc
@@ -4,6 +4,8 @@
 
 #include "ui/ozone/platform/x11/x11_screen_ozone.h"
 
+#include <memory>
+
 #include "testing/gtest/include/gtest/gtest.h"
 #include "ui/display/display.h"
 
diff --git a/ui/ozone/platform/x11/x11_surface_factory.cc b/ui/ozone/platform/x11/x11_surface_factory.cc
index 74aa63cec999..6011ca12beda 100644
--- a/ui/ozone/platform/x11/x11_surface_factory.cc
+++ b/ui/ozone/platform/x11/x11_surface_factory.cc
@@ -4,6 +4,8 @@
 
 #include "ui/ozone/platform/x11/x11_surface_factory.h"
 
+#include <memory>
+
 #include "gpu/vulkan/buildflags.h"
 #include "ui/gfx/x/x11.h"
 #include "ui/gfx/x/x11_types.h"
diff --git a/ui/ozone/platform/x11/x11_window_ozone_unittest.cc b/ui/ozone/platform/x11/x11_window_ozone_unittest.cc
index 2a745701007a..2484f5186989 100644
--- a/ui/ozone/platform/x11/x11_window_ozone_unittest.cc
+++ b/ui/ozone/platform/x11/x11_window_ozone_unittest.cc
@@ -4,6 +4,8 @@
 
 #include "ui/ozone/platform/x11/x11_window_ozone.h"
 
+#include <memory>
+
 #include "base/run_loop.h"
 #include "base/test/scoped_task_environment.h"
 #include "testing/gtest/include/gtest/gtest.h"
diff --git a/ui/ozone/platform_object_internal.h b/ui/ozone/platform_object_internal.h
index bcf1b46a95f8..c548931c5d69 100644
--- a/ui/ozone/platform_object_internal.h
+++ b/ui/ozone/platform_object_internal.h
@@ -5,6 +5,8 @@
 #ifndef UI_OZONE_PLATFORM_OBJECT_INTERNAL_H_
 #define UI_OZONE_PLATFORM_OBJECT_INTERNAL_H_
 
+#include <memory>
+
 #include "base/memory/ptr_util.h"
 #include "ui/ozone/ozone_export.h"
 #include "ui/ozone/platform_constructor_list.h"
diff --git a/ui/ozone/public/client_native_pixmap_factory_ozone.cc b/ui/ozone/public/client_native_pixmap_factory_ozone.cc
index 7d6428b48d8d..c118a3125bea 100644
--- a/ui/ozone/public/client_native_pixmap_factory_ozone.cc
+++ b/ui/ozone/public/client_native_pixmap_factory_ozone.cc
@@ -4,6 +4,8 @@
 
 #include "ui/ozone/public/client_native_pixmap_factory_ozone.h"
 
+#include <memory>
+
 #include "base/trace_event/trace_event.h"
 #include "ui/ozone/platform_object.h"
 #include "ui/ozone/platform_selection.h"
diff --git a/ui/ozone/public/client_native_pixmap_factory_ozone.h b/ui/ozone/public/client_native_pixmap_factory_ozone.h
index 744e868ad855..a1525893cc3f 100644
--- a/ui/ozone/public/client_native_pixmap_factory_ozone.h
+++ b/ui/ozone/public/client_native_pixmap_factory_ozone.h
@@ -5,6 +5,8 @@
 #ifndef UI_OZONE_PUBLIC_CLIENT_NATIVE_PIXMAP_FACTORY_OZONE_H_
 #define UI_OZONE_PUBLIC_CLIENT_NATIVE_PIXMAP_FACTORY_OZONE_H_
 
+#include <memory>
+
 #include "ui/gfx/client_native_pixmap_factory.h"
 #include "ui/ozone/ozone_export.h"
 
diff --git a/ui/ozone/public/input_controller.cc b/ui/ozone/public/input_controller.cc
index 4ba1521c5308..daacfbb44cde 100644
--- a/ui/ozone/public/input_controller.cc
+++ b/ui/ozone/public/input_controller.cc
@@ -4,6 +4,8 @@
 
 #include "ui/ozone/public/input_controller.h"
 
+#include <memory>
+
 #include "base/callback.h"
 #include "base/logging.h"
 #include "base/macros.h"
diff --git a/ui/ozone/public/overlay_plane.cc b/ui/ozone/public/overlay_plane.cc
index ddc0abbb60a0..35cd261609a8 100644
--- a/ui/ozone/public/overlay_plane.cc
+++ b/ui/ozone/public/overlay_plane.cc
@@ -4,6 +4,8 @@
 
 #include "ui/ozone/public/overlay_plane.h"
 
+#include <memory>
+
 namespace ui {
 
 OverlayPlane::OverlayPlane() {}
diff --git a/ui/ozone/public/overlay_plane.h b/ui/ozone/public/overlay_plane.h
index c1b5e8d2b22a..9689177ec303 100644
--- a/ui/ozone/public/overlay_plane.h
+++ b/ui/ozone/public/overlay_plane.h
@@ -5,6 +5,8 @@
 #ifndef UI_OZONE_PUBLIC_OVERLAY_PLANE_H_
 #define UI_OZONE_PUBLIC_OVERLAY_PLANE_H_
 
+#include <memory>
+
 #include "base/macros.h"
 #include "base/memory/ref_counted.h"
 #include "ui/gfx/geometry/rect.h"
diff --git a/ui/ozone/public/ozone_platform.cc b/ui/ozone/public/ozone_platform.cc
index aa2e5351d3bf..ce18a5bfc649 100644
--- a/ui/ozone/public/ozone_platform.cc
+++ b/ui/ozone/public/ozone_platform.cc
@@ -4,6 +4,7 @@
 
 #include "ui/ozone/public/ozone_platform.h"
 
+#include <memory>
 #include <utility>
 
 #include "base/logging.h"
diff --git a/ui/ozone/public/surface_factory_ozone.cc b/ui/ozone/public/surface_factory_ozone.cc
index a12ec481e51c..faa7bb627027 100644
--- a/ui/ozone/public/surface_factory_ozone.cc
+++ b/ui/ozone/public/surface_factory_ozone.cc
@@ -5,6 +5,7 @@
 #include "ui/ozone/public/surface_factory_ozone.h"
 
 #include <stdlib.h>
+#include <memory>
 
 #include "base/command_line.h"
 #include "gpu/vulkan/buildflags.h"
diff --git a/ui/ozone/public/surface_factory_ozone.h b/ui/ozone/public/surface_factory_ozone.h
index 6ac0ea495c4d..16758eb7fc8e 100644
--- a/ui/ozone/public/surface_factory_ozone.h
+++ b/ui/ozone/public/surface_factory_ozone.h
@@ -6,7 +6,6 @@
 #define UI_OZONE_PUBLIC_SURFACE_FACTORY_OZONE_H_
 
 #include <stdint.h>
-
 #include <memory>
 #include <vector>
 
diff --git a/ui/ozone/public/swap_completion_callback.h b/ui/ozone/public/swap_completion_callback.h
index c5be52508371..367e30688e3f 100644
--- a/ui/ozone/public/swap_completion_callback.h
+++ b/ui/ozone/public/swap_completion_callback.h
@@ -6,6 +6,7 @@
 #define UI_OZONE_PUBLIC_SWAP_COMPLETION_CALLBACK_H_
 
 #include <memory>
+
 #include "base/callback.h"
 #include "ui/gfx/swap_result.h"
 
-- 
2.20.1

